// @faclon-labs/design-sdk/dist/components/overlays/DropdownMenu/ActionListItem.js
import { jsx as i, jsxs as t } from "react/jsx-runtime";
import { cn as s } from "../../../utils/cn.js";
import { Divider as v } from "../../layout/Divider/Divider.js";
/* empty css                   */
const p = () => /* @__PURE__ */ i("svg", { width: "12", height: "12", viewBox: "0 0 12 12", fill: "none", xmlns: "http://www.w3.org/2000/svg", children: /* @__PURE__ */ i("path", { d: "M10 3L4.5 8.5L2 6", stroke: "currentColor", strokeWidth: "1.5", strokeLinecap: "round", strokeLinejoin: "round" }) });
function N({
  contentType: c = "Item",
  selectionType: f = "None",
  title: o,
  description: m,
  leadingIcon: r,
  trailing: e,
  badges: h,
  isSelected: a = !1,
  isDisabled: d = !1,
  isDestructive: _ = !1,
  className: l,
  ...n
}) {
  return c === "Separator" ? /* @__PURE__ */ i("div", { className: s("fds-action-list-item fds-action-list-item--separator", l), ...n, children: /* @__PURE__ */ i(v, { variant: "Muted" }) }) : c === "SectionHeading" ? /* @__PURE__ */ i("div", { className: s("fds-action-list-item fds-action-list-item--heading", l), ...n, children: /* @__PURE__ */ t("div", { className: "fds-action-list-item__heading-wrapper", children: [
    /* @__PURE__ */ i("div", { className: "fds-action-list-item__heading-body", children: /* @__PURE__ */ i("span", { className: "fds-action-list-item__heading-title BodySmallSemibold", children: o }) }),
    e && /* @__PURE__ */ i("div", { className: "fds-action-list-item__trailing", children: e })
  ] }) }) : /* @__PURE__ */ i(
    "div",
    {
      className: s(
        "fds-action-list-item",
        a && "fds-action-list-item--selected",
        d && "fds-action-list-item--disabled",
        _ && "fds-action-list-item--destructive",
        l
      ),
      role: "menuitem",
      "aria-disabled": d || void 0,
      tabIndex: d ? -1 : 0,
      ...n,
      children: /* @__PURE__ */ t("div", { className: "fds-action-list-item__wrapper", children: [
        /* @__PURE__ */ t("div", { className: "fds-action-list-item__body", children: [
          f === "Multiple" ? /* @__PURE__ */ i("span", { className: s("fds-action-list-item__checkbox", a && "fds-action-list-item__checkbox--checked"), children: a && /* @__PURE__ */ i(p, {}) }) : r ? /* @__PURE__ */ i("span", { className: "fds-action-list-item__leading", children: r }) : null,
          /* @__PURE__ */ t("div", { className: "fds-action-list-item__content", children: [
            /* @__PURE__ */ t("div", { className: "fds-action-list-item__title-row", children: [
              /* @__PURE__ */ i("span", { className: "fds-action-list-item__title BodyMediumRegular", children: o }),
              h
            ] }),
            m && /* @__PURE__ */ i("span", { className: "fds-action-list-item__description BodySmallRegular", children: m })
          ] })
        ] }),
        e && /* @__PURE__ */ i("div", { className: "fds-action-list-item__trailing", children: e })
      ] })
    }
  );
}
N.displayName = "ActionListItem";
export {
  N as ActionListItem
};
