// @faclon-labs/design-sdk/dist/components/overlays/DropdownMenu/DropdownFooter.js
import { jsxs as e, jsx as o, Fragment as a } from "react/jsx-runtime";
import { Divider as f } from "../../layout/Divider/Divider.js";
/* empty css                   */
function i({
  primaryAction: d,
  secondaryAction: r,
  children: s,
  stacking: t = "Horizontal"
}) {
  const n = d || r;
  return /* @__PURE__ */ e("div", { className: `fds-dropdown-footer fds-dropdown-footer--${t.toLowerCase()}`, children: [
    /* @__PURE__ */ o(f, { variant: "Muted" }),
    /* @__PURE__ */ e("div", { className: "fds-dropdown-footer__container", children: [
      s && /* @__PURE__ */ o("div", { className: "fds-dropdown-footer__slot", children: s }),
      n && /* @__PURE__ */ o("div", { className: "fds-dropdown-footer__actions", children: t === "Horizontal" ? /* @__PURE__ */ e(a, { children: [
        r && /* @__PURE__ */ o("div", { className: "fds-dropdown-footer__action", children: r }),
        d && /* @__PURE__ */ o("div", { className: "fds-dropdown-footer__action", children: d })
      ] }) : /* @__PURE__ */ e(a, { children: [
        d && /* @__PURE__ */ o("div", { className: "fds-dropdown-footer__action", children: d }),
        r && /* @__PURE__ */ o("div", { className: "fds-dropdown-footer__action", children: r })
      ] }) })
    ] })
  ] });
}
i.displayName = "DropdownFooter";
export {
  i as DropdownFooter
};
