// @faclon-labs/design-sdk/dist/components/overlays/DropdownMenu/DropdownHeader.js
import { jsxs as d, jsx as e } from "react/jsx-runtime";
import { Divider as i } from "../../layout/Divider/Divider.js";
/* empty css                   */
function l({
  title: o,
  subtitle: r,
  leadingItem: s,
  trailingItem: n,
  counter: a
}) {
  return /* @__PURE__ */ d("div", { className: "fds-dropdown-header", children: [
    /* @__PURE__ */ d("div", { className: "fds-dropdown-header__container", children: [
      /* @__PURE__ */ d("div", { className: "fds-dropdown-header__left", children: [
        s,
        /* @__PURE__ */ d("div", { className: "fds-dropdown-header__content", children: [
          /* @__PURE__ */ d("div", { className: "fds-dropdown-header__heading", children: [
            /* @__PURE__ */ e("span", { className: "fds-dropdown-header__title BodyLargeSemibold", children: o }),
            a && /* @__PURE__ */ e("span", { className: "fds-dropdown-header__counter", children: a })
          ] }),
          r && /* @__PURE__ */ e("span", { className: "fds-dropdown-header__subtitle BodySmallRegular", children: r })
        ] })
      ] }),
      n
    ] }),
    /* @__PURE__ */ e(i, { variant: "Muted" })
  ] });
}
l.displayName = "DropdownHeader";
export {
  l as DropdownHeader
};
