// @faclon-labs/design-sdk/dist/components/navigation/Tabs/TabItem.js
import { jsxs as u, jsx as c } from "react/jsx-runtime";
import { cn as d } from "../../../utils/cn.js";
/* empty css            */
const p = {
  Medium: "BodyMediumMedium",
  Large: "BodyLargeMedium"
};
function L({
  label: s,
  variant: o = "Bordered",
  size: a = "Medium",
  orientation: m = "Horizontal",
  isSelected: e = !1,
  isFullWidth: i = !1,
  isDisabled: t = !1,
  leadingItem: r,
  trailingItem: b,
  className: l,
  ...f
}) {
  const n = d(
    "fds-tab-item",
    `fds-tab-item--${o.toLowerCase()}`,
    `fds-tab-item--${a.toLowerCase()}`,
    `fds-tab-item--${m.toLowerCase()}`,
    e && "fds-tab-item--selected",
    i && "fds-tab-item--full-width",
    t && "fds-tab-item--disabled",
    l
  );
  return /* @__PURE__ */ u(
    "button",
    {
      type: "button",
      role: "tab",
      className: n,
      "aria-selected": e,
      "aria-disabled": t || void 0,
      disabled: t,
      tabIndex: e ? 0 : -1,
      ...f,
      children: [
        r,
        /* @__PURE__ */ c("span", { className: d("fds-tab-item__label", p[a]), children: s }),
        b
      ]
    }
  );
}
L.displayName = "TabItem";
export {
  L as TabItem
};
