// @faclon-labs/design-sdk/dist/components/surfaces/Accordion/AccordionItem.js
import { jsx as e, jsxs as s } from "react/jsx-runtime";
import { useState as B, useCallback as f } from "react";
import { useAccordionContext as g } from "./Accordion.js";
import { AccordionLeadingItem as L } from "./AccordionLeadingItem.js";
/* empty css                  */
function b() {
  return /* @__PURE__ */ e("svg", { viewBox: "0 0 9.33 5.33", fill: "none", xmlns: "http://www.w3.org/2000/svg", "aria-hidden": "true", children: /* @__PURE__ */ e(
    "path",
    {
      fillRule: "evenodd",
      clipRule: "evenodd",
      d: "M0.195 0.195C0.456-0.065 0.878-0.065 1.138 0.195L4.667 3.724L8.195 0.195C8.456-0.065 8.878-0.065 9.138 0.195C9.398 0.456 9.398 0.878 9.138 1.138L5.138 5.138C4.878 5.398 4.456 5.398 4.195 5.138L0.195 1.138C-0.065 0.878-0.065 0.456 0.195 0.195Z",
      fill: "currentColor"
    }
  ) });
}
function k({
  value: i,
  title: p,
  bodyText: r,
  children: d,
  leading: u = "None",
  leadingIcon: _,
  leadingNumber: h,
  isExpanded: m,
  defaultExpanded: y = !1,
  onExpandedChange: n,
  className: N
}) {
  const t = g(), [v, w] = B(y), a = m !== void 0, c = t && i ? t.expandedKeys.has(i) : a ? m : v, l = f(() => {
    if (t && i)
      t.toggleKey(i);
    else {
      const o = !c;
      a || w(o), n == null || n(o);
    }
  }, [t, i, c, a, n]), x = f(
    (o) => {
      (o.key === "Enter" || o.key === " ") && (o.preventDefault(), l());
    },
    [l]
  ), C = [
    "fds-accordion-item",
    c && "fds-accordion-item--expanded",
    N
  ].filter(Boolean).join(" ");
  return /* @__PURE__ */ e("div", { className: C, children: /* @__PURE__ */ s("div", { className: "fds-accordion-item__root", children: [
    /* @__PURE__ */ s(
      "button",
      {
        className: "fds-accordion-item__header",
        type: "button",
        onClick: l,
        onKeyDown: x,
        "aria-expanded": c,
        children: [
          /* @__PURE__ */ s("div", { className: "fds-accordion-item__title-area", children: [
            /* @__PURE__ */ e(
              L,
              {
                leading: u,
                icon: _,
                number: h
              }
            ),
            /* @__PURE__ */ e("span", { className: "fds-accordion-item__title BodyMediumMedium", children: p })
          ] }),
          /* @__PURE__ */ e("span", { className: "fds-accordion-item__chevron", children: /* @__PURE__ */ e(b, {}) })
        ]
      }
    ),
    (d || r) && /* @__PURE__ */ s("div", { className: "fds-accordion-item__body", role: "region", children: [
      r && !d && /* @__PURE__ */ e("p", { className: "fds-accordion-item__body-text BodyMediumRegular", children: r }),
      d
    ] })
  ] }) });
}
k.displayName = "AccordionItem";
export {
  k as AccordionItem
};
