// @faclon-labs/design-sdk/dist/components/product/ComparisonToggle/ComparisonButton.js
import { jsx as r, jsxs as a } from "react/jsx-runtime";
import { cn as i } from "../../../utils/cn.js";
/* empty css                     */
function s({ selected: t }) {
  return /* @__PURE__ */ a("svg", { width: "16", height: "16", viewBox: "0 0 16 16", fill: "none", xmlns: "http://www.w3.org/2000/svg", "aria-hidden": "true", children: [
    /* @__PURE__ */ r(
      "path",
      {
        d: "M5.33 6L8 3.33L10.67 6",
        stroke: t ? "var(--text-positive-default)" : "var(--text-default-tertiary)",
        strokeWidth: "1.5",
        strokeLinecap: "round",
        strokeLinejoin: "round"
      }
    ),
    /* @__PURE__ */ r(
      "path",
      {
        d: "M10.67 10L8 12.67L5.33 10",
        stroke: t ? "var(--text-error-default)" : "var(--text-default-tertiary)",
        strokeWidth: "1.5",
        strokeLinecap: "round",
        strokeLinejoin: "round"
      }
    )
  ] });
}
function d({
  isSelected: t = !1,
  isDisabled: o = !1,
  className: e,
  ...n
}) {
  return /* @__PURE__ */ r(
    "button",
    {
      type: "button",
      className: i(
        "fds-comparison-btn",
        t ? "fds-comparison-btn--selected" : "fds-comparison-btn--unselected",
        o && "fds-comparison-btn--disabled",
        e
      ),
      disabled: o,
      "aria-pressed": t,
      "aria-label": "Toggle comparison",
      ...n,
      children: /* @__PURE__ */ r(s, { selected: t })
    }
  );
}
d.displayName = "ComparisonButton";
export {
  d as ComparisonButton
};
