// @faclon-labs/design-sdk/dist/components/product/DatePicker/CalendarFooter.js
import { jsxs as a, jsx as r } from "react/jsx-runtime";
import { cn as c } from "../../../utils/cn.js";
import { Button as o } from "../../actions/Button/Button.js";
import { Divider as d } from "../../layout/Divider/Divider.js";
/* empty css                   */
function s({
  cancelLabel: i = "Cancel",
  applyLabel: e = "Apply",
  onCancel: l,
  onApply: n,
  className: t,
  ...m
}) {
  return /* @__PURE__ */ a("div", { className: c("fds-calendar-footer", t), ...m, children: [
    /* @__PURE__ */ r(d, { variant: "Muted" }),
    /* @__PURE__ */ a("div", { className: "fds-calendar-footer__actions", children: [
      /* @__PURE__ */ r(
        o,
        {
          variant: "Secondary",
          color: "Primary",
          size: "Small",
          label: i,
          onClick: l
        }
      ),
      /* @__PURE__ */ r(
        o,
        {
          variant: "Primary",
          color: "Primary",
          size: "Small",
          label: e,
          onClick: n
        }
      )
    ] })
  ] });
}
s.displayName = "CalendarFooter";
export {
  s as CalendarFooter
};
