// @faclon-labs/design-sdk/dist/components/product/DatePicker/CalendarHeader.js
import { jsxs as e, jsx as a } from "react/jsx-runtime";
import { ChevronDown as r, ChevronUp as t } from "react-feather";
import { cn as m } from "../../../utils/cn.js";
import { IconButton as l } from "../../actions/IconButton/IconButton.js";
/* empty css                   */
function h({
  label: i,
  onPrev: n,
  onNext: o,
  onLabelClick: d,
  className: s,
  ...c
}) {
  return /* @__PURE__ */ e("div", { className: m("fds-calendar-header", s), ...c, children: [
    /* @__PURE__ */ e(
      "button",
      {
        type: "button",
        className: "fds-calendar-header__label",
        onClick: d,
        children: [
          /* @__PURE__ */ a("span", { className: "fds-calendar-header__label-text BodyMediumMedium", children: i }),
          /* @__PURE__ */ a(r, { size: 16, className: "fds-calendar-header__label-icon" })
        ]
      }
    ),
    /* @__PURE__ */ e("div", { className: "fds-calendar-header__nav", children: [
      /* @__PURE__ */ a(
        l,
        {
          icon: /* @__PURE__ */ a(t, { size: 16 }),
          variant: "Tertiary",
          color: "Primary",
          size: "XSmall",
          onClick: n,
          "aria-label": "Previous month"
        }
      ),
      /* @__PURE__ */ a(
        l,
        {
          icon: /* @__PURE__ */ a(r, { size: 16 }),
          variant: "Tertiary",
          color: "Primary",
          size: "XSmall",
          onClick: o,
          "aria-label": "Next month"
        }
      )
    ] })
  ] });
}
h.displayName = "CalendarHeader";
export {
  h as CalendarHeader
};
