// @faclon-labs/design-sdk/dist/components/product/DatePicker/CalendarDayCell.js
import { jsxs as f, jsx as a } from "react/jsx-runtime";
import { cn as r } from "../../../utils/cn.js";
/* empty css                    */
function y({
  date: c,
  type: l = "default",
  isSelected: e = !1,
  isDisabled: d = !1,
  isCurrentDate: s = !1,
  className: t,
  ...n
}) {
  const o = l === "currentDate" || s;
  return /* @__PURE__ */ f(
    "button",
    {
      type: "button",
      className: r(
        "fds-day-cell",
        `fds-day-cell--${l}`,
        e && "fds-day-cell--selected",
        s && "fds-day-cell--today",
        d && "fds-day-cell--disabled",
        t
      ),
      disabled: d || l === "outOfMonth",
      "aria-selected": e,
      ...n,
      children: [
        /* @__PURE__ */ a("span", { className: "fds-day-cell__inner", children: /* @__PURE__ */ a("span", { className: "fds-day-cell__text BodyMediumRegular", children: c }) }),
        o && /* @__PURE__ */ a("span", { className: "fds-day-cell__dot" })
      ]
    }
  );
}
y.displayName = "CalendarDayCell";
export {
  y as CalendarDayCell
};
