// @faclon-labs/design-sdk/dist/components/product/DatePicker/CalendarBase.js
import { jsxs as m, jsx as a } from "react/jsx-runtime";
import { cn as n } from "../../../utils/cn.js";
import { CalendarHeader as S } from "./CalendarHeader.js";
import { CalendarWeekdays as x } from "./CalendarWeekdays.js";
import { CalendarDayCell as j } from "./CalendarDayCell.js";
import { CalendarFooter as v } from "./CalendarFooter.js";
import { MonthYearCell as B } from "./MonthYearCell.js";
/* empty css                 */
function M({
  view: r = "date",
  headerLabel: o = "March 2026",
  onPrev: c,
  onNext: p,
  onHeaderClick: f,
  days: b = [],
  onDayClick: s,
  items: _ = [],
  onItemClick: l,
  showFooter: h = !1,
  onCancel: N,
  onApply: $,
  className: C,
  ...u
}) {
  return /* @__PURE__ */ m("div", { className: n("fds-calendar-base", C), ...u, children: [
    /* @__PURE__ */ m("div", { className: "fds-calendar-base__header", children: [
      /* @__PURE__ */ a(
        S,
        {
          label: o,
          onPrev: c,
          onNext: p,
          onLabelClick: f
        }
      ),
      r === "date" && /* @__PURE__ */ a(x, {})
    ] }),
    /* @__PURE__ */ m("div", { className: n("fds-calendar-base__body", r !== "date" && "fds-calendar-base__body--grid"), children: [
      r === "date" && b.map((t, d) => /* @__PURE__ */ a("div", { id: `week-${d}`, className: "fds-calendar-base__week", children: t.map((e, i) => /* @__PURE__ */ a(
        j,
        {
          id: `${d}-${i}`,
          date: e.date,
          type: e.type,
          isSelected: e.isSelected,
          isCurrentDate: e.isCurrentDate,
          onClick: () => s == null ? void 0 : s(e)
        }
      )) })),
      r !== "date" && _.map((t, d) => /* @__PURE__ */ a("div", { id: `row-${d}`, className: "fds-calendar-base__item-row", children: t.map((e, i) => /* @__PURE__ */ a(
        B,
        {
          id: `${d}-${i}`,
          label: e.label,
          isSelected: e.isSelected,
          onClick: () => l == null ? void 0 : l(e)
        }
      )) }))
    ] }),
    h && /* @__PURE__ */ a(v, { onCancel: N, onApply: $ })
  ] });
}
M.displayName = "CalendarBase";
export {
  M as CalendarBase
};
