// @faclon-labs/design-sdk/dist/components/product/DatePicker/datePickerUtils.js
function b(r, n, t, a, e) {
  const o = new Date(r, n, 1).getDay(), f = new Date(r, n + 1, 0).getDate(), Y = new Date(r, n, 0).getDate(), _ = o + f, m = Math.ceil(_ / 7), i = /* @__PURE__ */ new Date(), v = `${i.getFullYear()}-${i.getMonth()}-${i.getDate()}`, h = [];
  let c = 1, T = 1;
  for (let D = 0; D < m; D++) {
    const u = [];
    for (let d = 0; d < 7; d++) {
      const p = D * 7 + d;
      if (p < o) {
        const w = Y - o + p + 1;
        u.push({ date: w, type: "outOfMonth" });
      } else if (c > f)
        u.push({ date: T++, type: "outOfMonth" });
      else {
        const w = new Date(r, n, c), y = `${r}-${n}-${c}` === v;
        let l = y ? "currentDate" : "default", M = !1;
        if (t && a) {
          const S = t.getTime(), $ = a.getTime(), g = w.getTime();
          g === S ? l = "rangeStart" : g === $ ? l = "rangeEnd" : g > S && g < $ && (l = "rangeIn");
        }
        e && e.getFullYear() === r && e.getMonth() === n && e.getDate() === c && (M = !0), u.push({ date: c, type: l, isSelected: M, isCurrentDate: y }), c++;
      }
    }
    h.push(u);
  }
  return h;
}
const F = ["Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"];
function x(r) {
  const n = [];
  for (let t = 0; t < 4; t++) {
    const a = [];
    for (let e = 0; e < 3; e++) {
      const s = t * 3 + e;
      a.push({ label: F[s], value: s, isSelected: s === r });
    }
    n.push(a);
  }
  return n;
}
function O(r, n) {
  const t = [];
  let a = r;
  for (let e = 0; e < 4; e++) {
    const s = [];
    for (let o = 0; o < 3; o++)
      s.push({ label: String(a), value: a, isSelected: a === n }), a++;
    t.push(s);
  }
  return t;
}
function k(r) {
  const n = /* @__PURE__ */ new Date(), t = new Date(n.getFullYear(), n.getMonth(), n.getDate()), a = new Date(t);
  switch (a.setDate(t.getDate() - 1), r) {
    case "today":
      return { start: t, end: t };
    case "yesterday":
      return { start: a, end: a };
    case "current_week": {
      const e = t.getDay(), s = new Date(t);
      return s.setDate(t.getDate() - (e + 6) % 7), { start: s, end: t };
    }
    case "previous_7_days": {
      const e = new Date(t);
      return e.setDate(t.getDate() - 7), { start: e, end: a };
    }
    case "current_month":
      return { start: new Date(t.getFullYear(), t.getMonth(), 1), end: t };
    case "previous_month": {
      const e = new Date(t.getFullYear(), t.getMonth() - 1, 1), s = new Date(t.getFullYear(), t.getMonth(), 0);
      return { start: e, end: s };
    }
    case "previous_3_month": {
      const e = new Date(t.getFullYear(), t.getMonth() - 3, 1), s = new Date(t.getFullYear(), t.getMonth(), 0);
      return { start: e, end: s };
    }
    case "previous_12_month": {
      const e = new Date(t.getFullYear(), t.getMonth() - 12, 1), s = new Date(t.getFullYear(), t.getMonth(), 0);
      return { start: e, end: s };
    }
    case "current_year":
      return { start: new Date(t.getFullYear(), 0, 1), end: t };
    case "previous_year": {
      const e = t.getFullYear() - 1;
      return { start: new Date(e, 0, 1), end: new Date(e, 11, 31) };
    }
    default:
      return null;
  }
}
function A(r) {
  const n = String(r.getDate()).padStart(2, "0"), t = String(r.getMonth() + 1).padStart(2, "0"), a = r.getFullYear();
  return `${n}/${t}/${a}`;
}
function H(r) {
  const n = String(r.getHours()).padStart(2, "0"), t = String(r.getMinutes()).padStart(2, "0");
  return `${n}:${t}`;
}
function I(r, n, t, a) {
  switch (r) {
    case "date":
      return `${F[t]} ${n}`;
    case "month":
      return String(n);
    case "year":
      return `${a} - ${a + 11}`;
  }
}
export {
  A as formatDate,
  H as formatTime,
  b as generateCalendarDays,
  x as generateMonths,
  O as generateYears,
  I as getHeaderLabel,
  k as getPresetDateRange
};
