// @faclon-labs/design-sdk/dist/components/product/DatePicker/DatePresetSidebar.js
import { jsx as l } from "react/jsx-runtime";
import { cn as v } from "../../../utils/cn.js";
import { DatePresetBase as i } from "./DatePresetBase.js";
/* empty css                      */
const s = [
  { label: "Custom", value: "custom" },
  { label: "Today", value: "today" },
  { label: "Yesterday", value: "yesterday" },
  { label: "Current Week", value: "current_week" },
  { label: "Previous 7 Days", value: "previous_7_days" },
  { label: "Current Month", value: "current_month" },
  { label: "Previous Month", value: "previous_month" },
  { label: "Previous 3 Month", value: "previous_3_month" },
  { label: "Previous 12 Month", value: "previous_12_month" },
  { label: "Current Year", value: "current_year" },
  { label: "Previous Year", value: "previous_year" }
];
function n({
  presets: r = s,
  selectedValue: u,
  onPresetSelect: a,
  className: o,
  ...t
}) {
  return /* @__PURE__ */ l("div", { className: v("fds-date-preset-sidebar", o), ...t, children: r.map((e) => /* @__PURE__ */ l(
    i,
    {
      id: e.value,
      label: e.label,
      isSelected: u === e.value,
      onClick: () => a == null ? void 0 : a(e.value)
    }
  )) });
}
n.displayName = "DatePresetSidebar";
export {
  n as DatePresetSidebar
};
