// @faclon-labs/design-sdk/dist/components/product/DatePicker/DatePickerPopover.js
import { jsxs as l, jsx as e } from "react/jsx-runtime";
import { cn as _ } from "../../../utils/cn.js";
import { TextInput as i } from "../../inputs/TextInput/TextInput.js";
import { Divider as w } from "../../layout/Divider/Divider.js";
import { CalendarHeader as z } from "./CalendarHeader.js";
import { CalendarWeekdays as A } from "./CalendarWeekdays.js";
import { CalendarDayCell as B } from "./CalendarDayCell.js";
import { CalendarFooter as G } from "./CalendarFooter.js";
import { MonthYearCell as J } from "./MonthYearCell.js";
import { DatePresetSidebar as K } from "./DatePresetSidebar.js";
/* empty css                      */
function O({
  showPresets: u = !0,
  presets: N,
  selectedPreset: b,
  onPresetSelect: k,
  startDate: M = "",
  startTime: Y = "",
  endDate: P = "",
  endTime: $ = "",
  onStartDateChange: o,
  onStartTimeChange: c,
  onEndDateChange: s,
  onEndTimeChange: t,
  calendarLabel: h = "March 2026",
  onPrevMonth: x,
  onNextMonth: H,
  onCalendarLabelClick: j,
  view: p = "date",
  days: D = [],
  onDayClick: v,
  items: F = [],
  onItemClick: m,
  onCancel: L,
  onApply: V,
  className: W,
  ...q
}) {
  return /* @__PURE__ */ l("div", { className: _("fds-datepicker-popover", W), ...q, children: [
    u && /* @__PURE__ */ e(
      K,
      {
        presets: N,
        selectedValue: b,
        onPresetSelect: k
      }
    ),
    /* @__PURE__ */ l("div", { className: "fds-datepicker-popover__panel", children: [
      /* @__PURE__ */ l("div", { className: "fds-datepicker-popover__inputs", children: [
        /* @__PURE__ */ l("div", { className: "fds-datepicker-popover__input-col", children: [
          /* @__PURE__ */ e(
            i,
            {
              label: "Start Date",
              placeholder: "DD/MM/YYYY",
              value: M,
              onChange: (r) => o == null ? void 0 : o(r.value)
            }
          ),
          /* @__PURE__ */ e(
            i,
            {
              label: "End Date",
              placeholder: "DD/MM/YYYY",
              value: P,
              onChange: (r) => s == null ? void 0 : s(r.value)
            }
          )
        ] }),
        /* @__PURE__ */ l("div", { className: "fds-datepicker-popover__input-col", children: [
          /* @__PURE__ */ e(
            i,
            {
              label: "Start Time",
              placeholder: "HH:MM",
              value: Y,
              onChange: (r) => c == null ? void 0 : c(r.value)
            }
          ),
          /* @__PURE__ */ e(
            i,
            {
              label: "End Time",
              placeholder: "HH:MM",
              value: $,
              onChange: (r) => t == null ? void 0 : t(r.value)
            }
          )
        ] })
      ] }),
      /* @__PURE__ */ e(w, { variant: "Muted" }),
      /* @__PURE__ */ l("div", { className: "fds-datepicker-popover__calendar", children: [
        /* @__PURE__ */ l("div", { className: "fds-datepicker-popover__calendar-header", children: [
          /* @__PURE__ */ e(
            z,
            {
              label: h,
              onPrev: x,
              onNext: H,
              onLabelClick: j
            }
          ),
          p === "date" && /* @__PURE__ */ e(A, {})
        ] }),
        /* @__PURE__ */ l("div", { className: _("fds-datepicker-popover__calendar-body", p !== "date" && "fds-datepicker-popover__calendar-body--grid"), children: [
          p === "date" && D.map((r, a) => /* @__PURE__ */ e("div", { id: `week-${a}`, className: "fds-datepicker-popover__week", children: r.map((d, f) => /* @__PURE__ */ e(
            B,
            {
              id: `${a}-${f}`,
              date: d.date,
              type: d.type,
              isSelected: d.isSelected,
              isCurrentDate: d.isCurrentDate,
              onClick: () => v == null ? void 0 : v(d)
            }
          )) })),
          p !== "date" && F.map((r, a) => /* @__PURE__ */ e("div", { id: `row-${a}`, className: "fds-datepicker-popover__item-row", children: r.map((d, f) => /* @__PURE__ */ e(
            J,
            {
              id: `${a}-${f}`,
              label: d.label,
              isSelected: d.isSelected,
              onClick: () => m == null ? void 0 : m(d)
            }
          )) }))
        ] })
      ] }),
      /* @__PURE__ */ e(G, { onCancel: L, onApply: V })
    ] })
  ] });
}
O.displayName = "DatePickerPopover";
export {
  O as DatePickerPopover
};
