// @faclon-labs/design-sdk/dist/components/product/DatePicker/DatePicker.js
import { jsxs as g, jsx as P } from "react/jsx-runtime";
import { useState as l, useCallback as d, useRef as vt, useMemo as b } from "react";
import { cn as V } from "../../../utils/cn.js";
import { useClickOutside as Dt } from "../../../hooks/useClickOutside.js";
import { useKeyboard as Mt } from "../../../hooks/useKeyboard.js";
import { DatePickerTrigger as Ct } from "./DatePickerTrigger.js";
import { DatePickerPopover as St } from "./DatePickerPopover.js";
import { CalendarBase as wt } from "./CalendarBase.js";
import { generateCalendarDays as Tt, generateMonths as Yt, generateYears as Nt, getPresetDateRange as It, formatDate as v, formatTime as tt, getHeaderLabel as Lt } from "./datePickerUtils.js";
/* empty css               */
const bt = {
  custom: "Custom",
  today: "Today",
  yesterday: "Yesterday",
  current_week: "Current Week",
  previous_7_days: "Past 7 days",
  current_month: "Current Month",
  previous_month: "Previous Month",
  previous_3_month: "Previous 3 Month",
  previous_12_month: "Previous 12 Month",
  current_year: "Current Year",
  previous_year: "Previous Year"
};
function xt({
  mode: r = "single",
  isOpen: Y,
  onOpenChange: D,
  value: p,
  onChange: M,
  rangeValue: e,
  onRangeChange: C,
  presets: et,
  selectedPreset: x,
  onPresetSelect: N,
  label: st,
  placeholder: rt,
  helpText: ot,
  errorText: nt,
  validationState: it,
  isDisabled: E,
  showPeriodicity: A = !1,
  periodicitySlot: I,
  className: ct,
  ...lt
}) {
  const [ft, dt] = l(!1), y = Y ?? ft, a = d(
    (t) => {
      Y === void 0 && dt(t), D == null || D(t);
    },
    [Y, D]
  ), F = /* @__PURE__ */ new Date(), h = r === "single" ? p : e == null ? void 0 : e.start, [i, k] = l((h == null ? void 0 : h.getMonth()) ?? F.getMonth()), [m, f] = l((h == null ? void 0 : h.getFullYear()) ?? F.getFullYear()), [s, _] = l("date"), L = Math.floor(m / 12) * 12, [o, S] = l((e == null ? void 0 : e.start) ?? null), [c, u] = l((e == null ? void 0 : e.end) ?? null), [H, w] = l(e ? 2 : 0), [at, j] = l(x ?? "custom"), B = x ?? at, [T, $] = l(p ?? null), K = vt(null);
  Dt(K, () => {
    y && a(!1);
  }), Mt("Escape", () => {
    y && a(!1);
  });
  const R = b(
    () => Tt(
      m,
      i,
      r === "range" ? o : null,
      r === "range" ? c : null,
      r === "single" ? T : null
    ),
    [m, i, o, c, T, r]
  ), W = b(() => Yt(i), [i]), q = b(() => Nt(L, m), [L, m]), z = Lt(s, m, i, L), G = d(() => {
    s === "date" ? i === 0 ? (k(11), f((t) => t - 1)) : k((t) => t - 1) : f(s === "month" ? (t) => t - 1 : (t) => t - 12);
  }, [s, i]), J = d(() => {
    s === "date" ? i === 11 ? (k(0), f((t) => t + 1)) : k((t) => t + 1) : f(s === "month" ? (t) => t + 1 : (t) => t + 12);
  }, [s, i]), O = d(() => {
    s === "date" ? _("month") : s === "month" && _("year");
  }, [s]), Q = d(
    (t) => {
      s === "year" ? (f(t.value), _("month")) : s === "month" && (k(t.value), _("date"));
    },
    [s]
  ), U = (t) => {
    if (t.type === "outOfMonth") return;
    const n = new Date(m, i, t.date);
    r === "single" ? $(n) : H === 0 || H === 2 ? (S(n), u(null), w(1), j("custom")) : (n < o ? (S(n), u(o)) : u(n), w(2));
  }, mt = d(
    (t) => {
      N && N(t), j(t);
      const n = It(t);
      n && (S(n.start), u(n.end), w(2), k(n.start.getMonth()), f(n.start.getFullYear()));
    },
    [N]
  ), X = d(() => {
    r === "single" ? M == null || M(T) : o && c && (C == null || C({ start: o, end: c })), a(!1);
  }, [r, T, o, c, M, C, a]), Z = d(() => {
    r === "single" ? $(p ?? null) : (S((e == null ? void 0 : e.start) ?? null), u((e == null ? void 0 : e.end) ?? null), w(e ? 2 : 0)), _("date"), a(!1);
  }, [r, p, e, a]), pt = r === "single" && p ? v(p) : void 0, ht = r === "range" && e ? `${v(e.start)} - ${v(e.end)}` : void 0, kt = bt[B] ?? "Custom", yt = o ? v(o) : "", _t = c ? v(c) : "", ut = o ? tt(o) : "00:00", Pt = c ? tt(c) : "00:00";
  return /* @__PURE__ */ g("div", { ref: K, className: V("fds-datepicker", ct), ...lt, children: [
    /* @__PURE__ */ g("div", { className: V("fds-datepicker__trigger-row", A && !!I && "fds-datepicker__trigger-row--with-periodicity"), children: [
      /* @__PURE__ */ P(
        Ct,
        {
          selectionType: r,
          label: st,
          placeholder: rt,
          date: pt,
          presetValue: kt,
          range: ht,
          isOpen: y,
          isDisabled: E,
          helpText: ot,
          errorText: nt,
          validationState: it,
          onClick: () => !E && a(!y)
        }
      ),
      A && I && /* @__PURE__ */ P("div", { className: "fds-datepicker__periodicity", children: I })
    ] }),
    y && /* @__PURE__ */ P("div", { className: "fds-datepicker__popover", children: r === "single" ? /* @__PURE__ */ P(
      wt,
      {
        view: s,
        headerLabel: z,
        days: R,
        items: s === "month" ? W : q,
        onDayClick: U,
        onItemClick: Q,
        onHeaderClick: O,
        onPrev: G,
        onNext: J,
        showFooter: !0,
        onCancel: Z,
        onApply: X
      }
    ) : /* @__PURE__ */ P(
      St,
      {
        showPresets: !0,
        presets: et,
        selectedPreset: B,
        onPresetSelect: mt,
        startDate: yt,
        endDate: _t,
        startTime: ut,
        endTime: Pt,
        calendarLabel: z,
        view: s,
        days: R,
        items: s === "month" ? W : q,
        onDayClick: U,
        onItemClick: Q,
        onCalendarLabelClick: O,
        onPrevMonth: G,
        onNextMonth: J,
        onCancel: Z,
        onApply: X
      }
    ) })
  ] });
}
xt.displayName = "DatePicker";
export {
  xt as DatePicker
};
