// @faclon-labs/design-sdk/dist/components/product/DatePicker/DatePickerTrigger.js
import { jsxs as r, jsx as e } from "react/jsx-runtime";
import { forwardRef as C } from "react";
import { Calendar as m, ChevronDown as R } from "react-feather";
import { cn as d } from "../../../utils/cn.js";
import { InputFieldHeader as k } from "../../forms/InputFieldHeader/InputFieldHeader.js";
import { InputFieldFooter as y } from "../../forms/InputFieldFooter/InputFieldFooter.js";
/* empty css                      */
const z = C(
  ({
    selectionType: a = "single",
    label: s,
    placeholder: i = "Select Date",
    date: g,
    presetValue: p,
    range: l,
    isOpen: o = !1,
    isDisabled: n = !1,
    helpText: u,
    errorText: _,
    validationState: h = "none",
    onClick: N,
    onPresetClick: x,
    className: v
  }, b) => {
    const t = h === "error", f = t ? _ : u, c = a === "single" ? !!g : !!l;
    return /* @__PURE__ */ r(
      "div",
      {
        className: d(
          "fds-date-trigger",
          a === "range" && "fds-date-trigger--range",
          o && "fds-date-trigger--open",
          n && "fds-date-trigger--disabled",
          t && "fds-date-trigger--error",
          v
        ),
        children: [
          s && /* @__PURE__ */ e(k, { label: s }),
          /* @__PURE__ */ e(
            "button",
            {
              ref: b,
              type: "button",
              className: "fds-date-trigger__field",
              onClick: N,
              disabled: n,
              "aria-expanded": o,
              "aria-haspopup": "dialog",
              children: a === "range" ? (
                /* ── Range trigger ────────────────────────────────── */
                /* @__PURE__ */ r("span", { className: "fds-date-trigger__leading", children: [
                  /* @__PURE__ */ r(
                    "span",
                    {
                      className: "fds-date-trigger__preset",
                      onClick: x,
                      role: "button",
                      tabIndex: -1,
                      children: [
                        /* @__PURE__ */ e(m, { size: 12 }),
                        /* @__PURE__ */ e("span", { className: "fds-date-trigger__preset-label BodyMediumRegular", children: p || "Custom" }),
                        /* @__PURE__ */ e(R, { size: 12 })
                      ]
                    }
                  ),
                  /* @__PURE__ */ e(
                    "span",
                    {
                      className: d(
                        "fds-date-trigger__value BodyMediumRegular",
                        !c && "fds-date-trigger__value--placeholder"
                      ),
                      children: l || i
                    }
                  )
                ] })
              ) : (
                /* ── Single date trigger ──────────────────────────── */
                /* @__PURE__ */ r("span", { className: "fds-date-trigger__leading", children: [
                  /* @__PURE__ */ e("span", { className: "fds-date-trigger__icon", children: /* @__PURE__ */ e(m, { size: 16 }) }),
                  /* @__PURE__ */ e(
                    "span",
                    {
                      className: d(
                        "fds-date-trigger__value BodyMediumRegular",
                        !c && "fds-date-trigger__value--placeholder"
                      ),
                      children: g || i
                    }
                  )
                ] })
              )
            }
          ),
          f && /* @__PURE__ */ e(
            y,
            {
              helpText: f,
              state: t ? "error" : "default"
            }
          )
        ]
      }
    );
  }
);
z.displayName = "DatePickerTrigger";
export {
  z as DatePickerTrigger
};
