// @faclon-labs/design-sdk/dist/components/inputs/SelectInput/SelectInput.js
import { jsx as e, jsxs as s } from "react/jsx-runtime";
import { forwardRef as I, useId as S } from "react";
import { ChevronDown as j } from "react-feather";
import { cn as r } from "../../../utils/cn.js";
import { Tag as z } from "../../data-display/Tag/Tag.js";
import { InputFieldHeader as C } from "../../forms/InputFieldHeader/InputFieldHeader.js";
import { InputFieldFooter as T } from "../../forms/InputFieldFooter/InputFieldFooter.js";
/* empty css                */
const D = I(
  ({
    label: v,
    name: b,
    placeholder: c = "Select",
    value: n,
    tags: t,
    maxVisibleTags: u = 2,
    helpText: w,
    errorText: y,
    validationState: g = "none",
    isDisabled: o = !1,
    isOpen: l = !1,
    leadingIcon: f,
    leadingText: m,
    children: _,
    onClick: M,
    className: R
  }, B) => {
    const h = S(), i = t != null, a = i ? t.length > 0 : n != null && n !== "", d = g === "error", x = d ? y : w, F = i ? t.slice(0, u) : [], N = i ? t.length - u : 0;
    return /* @__PURE__ */ e("div", { className: r("fds-select-input__root", R), children: /* @__PURE__ */ s(
      "div",
      {
        className: r(
          "fds-text-input",
          o && "fds-text-input--disabled",
          d && "fds-text-input--error",
          l && "fds-select-input--open"
        ),
        children: [
          /* @__PURE__ */ e(C, { label: v, htmlFor: h }),
          /* @__PURE__ */ s("div", { className: "fds-text-input__field-wrapper fds-select-input__field-wrapper", children: [
            /* @__PURE__ */ s(
              "button",
              {
                ref: B,
                id: h,
                type: "button",
                role: "combobox",
                "aria-expanded": l,
                "aria-haspopup": "listbox",
                "aria-disabled": o || void 0,
                disabled: o,
                name: b,
                className: "fds-text-input__field fds-select-input__trigger",
                onClick: M,
                children: [
                  /* @__PURE__ */ s("span", { className: "fds-text-input__leading", children: [
                    f && /* @__PURE__ */ e("span", { className: "fds-text-input__icon", children: f }),
                    m && /* @__PURE__ */ e("span", { className: "fds-text-input__prefix BodyMediumRegular", children: m }),
                    i ? a ? /* @__PURE__ */ s("span", { className: "fds-select-input__tags", children: [
                      F.map((p) => /* @__PURE__ */ e(
                        z,
                        {
                          id: p.label,
                          label: p.label,
                          size: "Medium",
                          onDismiss: p.onDismiss
                        }
                      )),
                      N > 0 && /* @__PURE__ */ s("span", { className: "fds-select-input__overflow BodySmallRegular", children: [
                        "+",
                        N,
                        " more"
                      ] })
                    ] }) : /* @__PURE__ */ e("span", { className: "fds-select-input__value fds-select-input__value--placeholder BodyMediumRegular", children: c }) : /* @__PURE__ */ e(
                      "span",
                      {
                        className: r(
                          "fds-select-input__value BodyMediumRegular",
                          !a && "fds-select-input__value--placeholder"
                        ),
                        children: a ? n : c
                      }
                    )
                  ] }),
                  /* @__PURE__ */ e("span", { className: "fds-text-input__trailing", children: /* @__PURE__ */ e("span", { className: r("fds-select-input__chevron", l && "fds-select-input__chevron--open"), children: /* @__PURE__ */ e(j, { size: 16 }) }) })
                ]
              }
            ),
            l && _ && /* @__PURE__ */ e("div", { className: "fds-select-input__popover", children: _ })
          ] }),
          x && /* @__PURE__ */ e(
            T,
            {
              helpText: x,
              state: d ? "error" : "default"
            }
          )
        ]
      }
    ) });
  }
);
D.displayName = "SelectInput";
export {
  D as SelectInput
};
