// @faclon-labs/design-sdk/dist/components/inputs/TextInput/TextInput.js
import { jsxs as v, jsx as e } from "react/jsx-runtime";
import { forwardRef as ne, useId as ue, useState as z, useCallback as _ } from "react";
import { X as ce } from "react-feather";
import { Spinner as pe } from "../../feedback/Spinner/Spinner.js";
import { InputFieldHeader as me } from "../../forms/InputFieldHeader/InputFieldHeader.js";
import { InputFieldFooter as fe } from "../../forms/InputFieldFooter/InputFieldFooter.js";
/* empty css              */
const ve = {
  name: "name",
  email: "email",
  countryName: "country-name",
  postalCode: "postal-code",
  telephone: "tel",
  username: "username",
  none: "off"
}, _e = {
  text: "text",
  search: "search",
  telephone: "tel",
  email: "email",
  url: "url",
  number: "number"
}, xe = ne(
  ({
    label: n,
    labelPosition: O = "top",
    necessityIndicator: R,
    name: U,
    type: D = "text",
    placeholder: S,
    value: x,
    defaultValue: E,
    validationState: b = "none",
    helpText: X,
    errorText: M,
    successText: Y,
    onChange: s,
    onFocus: u,
    onBlur: c,
    isDisabled: G = !1,
    isRequired: p = !1,
    icon: T,
    showClearButton: P = !1,
    onClearButtonClicked: m,
    prefix: F,
    suffix: N,
    isLoading: d = !1,
    maxCharacters: i,
    autoFocus: J = !1,
    autoCompleteSuggestionType: $,
    keyboardReturnKeyType: j,
    className: K,
    id: Q,
    disabled: W,
    ...Z
  }, L) => {
    const B = ue(), h = Q ?? B, A = `${h}-help`, t = U ?? "", f = G || W || !1, g = S ?? "Enter value", l = R ?? (p ? "required" : void 0), [k, C] = z(!1), [ee, H] = z(E ?? ""), r = x !== void 0, y = r ? x : ee, a = b === "error" || (l === "required" || p) && k && !y, I = b === "success" && !a, V = a ? b === "error" ? M ?? "Error" : M ?? `${n} is required` : I ? Y ?? "" : X ?? "", te = a ? "error" : I ? "success" : "default", w = !d && (!!V || i !== void 0), se = y.length, ie = i !== void 0 ? `${se}/${i}` : "", re = _((o) => {
      const q = o.target.value;
      i !== void 0 && q.length > i || (r || H(q), s == null || s({ name: t, value: q }));
    }, [r, t, s, i]), de = _((o) => {
      C(!0), c == null || c({ name: t, value: o.target.value });
    }, [t, c]), le = _((o) => {
      u == null || u({ name: t, value: o.target.value });
    }, [t, u]), ae = _(() => {
      r || H(""), m == null || m(), s == null || s({ name: t, value: "" });
    }, [r, t, m, s]), oe = [
      "fds-text-input",
      O === "left" && "fds-text-input--label-left",
      f && "fds-text-input--disabled",
      d && "fds-text-input--loading",
      a && "fds-text-input--error",
      I && "fds-text-input--success",
      K
    ].filter(Boolean).join(" ");
    return /* @__PURE__ */ v("div", { className: oe, children: [
      /* @__PURE__ */ e(
        me,
        {
          label: l === "optional" ? `${n} (optional)` : n,
          necessityIndicator: l === "required" ? "required" : "none",
          htmlFor: h
        }
      ),
      /* @__PURE__ */ e("div", { className: "fds-text-input__field-wrapper", children: /* @__PURE__ */ v("div", { className: "fds-text-input__field", children: [
        /* @__PURE__ */ v("div", { className: "fds-text-input__leading", children: [
          T && /* @__PURE__ */ e("span", { className: "fds-text-input__icon", children: T }),
          F && /* @__PURE__ */ e("span", { className: "fds-text-input__prefix BodyMediumRegular", children: F }),
          /* @__PURE__ */ e(
            "input",
            {
              ref: L,
              className: "fds-text-input__input BodyMediumRegular",
              type: _e[D],
              id: h,
              name: t || void 0,
              placeholder: g,
              value: r ? x : void 0,
              defaultValue: r ? void 0 : E,
              disabled: f,
              required: l === "required" || p || void 0,
              autoFocus: J || void 0,
              autoComplete: $ ? ve[$] : void 0,
              enterKeyHint: j !== "default" ? j : void 0,
              maxLength: i,
              "aria-label": n,
              "aria-required": l === "required" || p || void 0,
              "aria-disabled": f || void 0,
              "aria-invalid": a || void 0,
              "aria-describedby": w ? A : void 0,
              "aria-busy": d || void 0,
              onChange: re,
              onBlur: de,
              onFocus: le,
              ...Z
            }
          )
        ] }),
        (N || P || d) && /* @__PURE__ */ v("span", { className: "fds-text-input__trailing", children: [
          N && /* @__PURE__ */ e("span", { className: "fds-text-input__suffix BodyMediumRegular", children: N }),
          P && y && !f && !d && /* @__PURE__ */ e(
            "button",
            {
              type: "button",
              className: "fds-text-input__clear",
              onClick: ae,
              "aria-label": "Clear input",
              tabIndex: -1,
              children: /* @__PURE__ */ e(ce, { size: 16 })
            }
          ),
          d && /* @__PURE__ */ e(pe, { color: "Brand", size: "Medium", accessibilityLabel: "Loading" })
        ] })
      ] }) }),
      w && /* @__PURE__ */ e(
        fe,
        {
          helpText: V || void 0,
          counterText: i !== void 0 ? ie : void 0,
          state: te,
          id: A
        }
      )
    ] });
  }
);
xe.displayName = "TextInput";
export {
  xe as TextInput
};
