// @faclon-labs/design-sdk/dist/components/overlays/Modal/Modal.js
import { jsx as d, jsxs as u } from "react/jsx-runtime";
import { useCallback as b } from "react";
import { X as _ } from "react-feather";
import { cn as r } from "../../../utils/cn.js";
import { useKeyboard as k } from "../../../hooks/useKeyboard.js";
/* empty css          */
function y({
  size: c = "Small",
  isOpen: m = !1,
  hasBodyPadding: f = !0,
  header: t,
  footer: o,
  children: i,
  onClose: a,
  className: e,
  ...s
}) {
  const p = b(
    (l) => {
      l.target === l.currentTarget && (a == null || a());
    },
    [a]
  );
  return k("Escape", () => a == null ? void 0 : a(), m), m ? /* @__PURE__ */ d(
    "div",
    {
      className: r("fds-modal__backdrop", e),
      onClick: p,
      role: "dialog",
      "aria-modal": "true",
      ...s,
      children: /* @__PURE__ */ u("div", { className: r("fds-modal", `fds-modal--${c.toLowerCase()}`), children: [
        t,
        !t && a && /* @__PURE__ */ d(
          "button",
          {
            type: "button",
            className: "fds-modal__floating-close",
            onClick: a,
            "aria-label": "Close",
            children: /* @__PURE__ */ d(_, { size: 20 })
          }
        ),
        i && /* @__PURE__ */ d("div", { className: r("fds-modal__body", f && "fds-modal__body--padded"), children: i }),
        o
      ] })
    }
  ) : null;
}
y.displayName = "Modal";
export {
  y as Modal
};
