// @faclon-labs/design-sdk/dist/components/actions/LinkButton/LinkButton.js
import { jsx as a, jsxs as M, Fragment as h } from "react/jsx-runtime";
import { forwardRef as L } from "react";
import { cn as N } from "../../../utils/cn.js";
/* empty css               */
const _ = {
  Large: "BodyLargeMedium",
  Medium: "BodyMediumMedium",
  Small: "BodySmallMedium",
  XSmall: "BodyXSmallMedium"
}, k = {
  Large: "BodyLargeSemibold",
  Medium: "BodyMediumSemibold",
  Small: "BodySmallSemibold",
  XSmall: "BodyXSmallSemibold"
}, A = L(
  ({
    type: l = "Anchor",
    color: t = "Primary",
    size: c = "Large",
    label: u,
    leadingIcon: e,
    trailingIcon: o,
    isDisabled: d = !1,
    className: f,
    children: p,
    href: y,
    target: S,
    rel: b,
    onClick: i,
    "aria-label": m
  }, s) => {
    const B = l === "Action" ? k : _, r = N(
      "fds-link",
      `fds-link--type-${l.toLowerCase()}`,
      `fds-link--color-${t.toLowerCase()}`,
      B[c],
      f
    ), n = /* @__PURE__ */ M(h, { children: [
      e && /* @__PURE__ */ a("span", { className: "fds-link__icon", children: e }),
      /* @__PURE__ */ a("span", { className: "fds-link__label", children: u ?? p }),
      o && /* @__PURE__ */ a("span", { className: "fds-link__icon", children: o })
    ] });
    return l === "Action" ? /* @__PURE__ */ a(
      "button",
      {
        ref: s,
        className: r,
        type: "button",
        disabled: d,
        "aria-disabled": d || void 0,
        "aria-label": m,
        onClick: i,
        children: n
      }
    ) : /* @__PURE__ */ a(
      "a",
      {
        ref: s,
        className: r,
        href: y,
        target: S,
        rel: b,
        "aria-label": m,
        onClick: i,
        children: n
      }
    );
  }
);
A.displayName = "LinkButton";
export {
  A as LinkButton
};
