// @faclon-labs/design-sdk/dist/components/inputs/Radio/RadioGroup.js
import { jsx as r, jsxs as p } from "react/jsx-runtime";
import { useContext as f, createContext as x } from "react";
/* empty css               */
const e = x(null);
function G() {
  return f(e);
}
const S = {
  Small: "BodySmallSemibold",
  Medium: "BodySmallSemibold",
  Large: "BodyMediumSemibold"
};
function g({
  label: o,
  name: a,
  value: i,
  onChange: d,
  size: l = "Small",
  isDisabled: t = !1,
  children: s,
  className: n,
  orientation: u = "Vertical"
}) {
  const m = {
    name: a,
    value: i,
    onChange: d,
    isDisabled: t
  }, c = [
    "fds-radio-group",
    `fds-radio-group--${u.toLowerCase()}`,
    n
  ].filter(Boolean).join(" ");
  return /* @__PURE__ */ r(e.Provider, { value: m, children: /* @__PURE__ */ p("div", { role: "radiogroup", className: c, "aria-label": o, children: [
    o && /* @__PURE__ */ r("span", { className: `fds-radio-group__label ${S[l]}`, children: o }),
    /* @__PURE__ */ r("div", { className: "fds-radio-group__body", children: s })
  ] }) });
}
g.displayName = "RadioGroup";
export {
  g as RadioGroup,
  G as useRadioGroupContext
};
