// @faclon-labs/design-sdk/dist/components/inputs/Radio/Radio.js
import { jsxs as S, jsx as r } from "react/jsx-runtime";
import { forwardRef as x, useId as B, useCallback as l } from "react";
import { useRadioGroupContext as C } from "./RadioGroup.js";
/* empty css          */
const j = {
  Small: "BodySmallRegular",
  Medium: "BodyMediumRegular",
  Large: "BodyLargeRegular"
}, v = x(
  ({
    label: c,
    size: t = "Small",
    isDisabled: m = !1,
    className: f,
    id: p,
    disabled: b,
    value: a,
    checked: h,
    onChange: s,
    name: u,
    ...y
  }, R) => {
    const k = B(), n = p ?? k, e = C(), o = m || b || (e == null ? void 0 : e.isDisabled) || !1, N = u ?? (e == null ? void 0 : e.name), _ = e ? e.value === a : h, g = l(
      (d) => {
        e && a !== void 0 && e.onChange(String(a)), s == null || s(d);
      },
      [e, a, s]
    ), w = l(
      (d) => {
        if (d.key === "Enter") {
          d.preventDefault();
          const i = d.currentTarget;
          i.checked || (i.checked = !0, i.dispatchEvent(new Event("change", { bubbles: !0 })), e && a !== void 0 && e.onChange(String(a)));
        }
      },
      [e, a]
    ), D = [
      "fds-radio",
      `fds-radio--size-${t.toLowerCase()}`,
      o && "fds-radio--disabled",
      f
    ].filter(Boolean).join(" ");
    return /* @__PURE__ */ S("label", { className: D, htmlFor: n, children: [
      /* @__PURE__ */ r(
        "input",
        {
          ref: R,
          className: "fds-radio__input",
          type: "radio",
          id: n,
          name: N,
          value: a,
          checked: _,
          disabled: o,
          "aria-disabled": o || void 0,
          onChange: g,
          onKeyDown: w,
          ...y
        }
      ),
      /* @__PURE__ */ r("span", { className: "fds-radio__circle", "aria-hidden": "true" }),
      /* @__PURE__ */ r("span", { className: `fds-radio__label ${j[t]}`, children: c })
    ] });
  }
);
v.displayName = "Radio";
export {
  v as Radio
};
