// @faclon-labs/design-sdk/dist/components/inputs/AutocompleteInput/AutocompleteInput.js
import { jsx as l, jsxs as g } from "react/jsx-runtime";
import { forwardRef as k } from "react";
import { TextInput as q } from "../TextInput/TextInput.js";
import { MultiSelectInput as z } from "../MultiSelectInput/MultiSelectInput.js";
import { cn as E } from "../../../utils/cn.js";
/* empty css                      */
const G = k(
  ({
    type: t = "single",
    label: a,
    name: i,
    placeholder: s = "Search",
    inputValue: x,
    onInputChange: e,
    helpText: c,
    errorText: d,
    validationState: r = "none",
    isDisabled: u = !1,
    isOpen: f = !1,
    icon: v,
    prefix: A,
    children: p,
    onFocus: m,
    onBlur: o,
    className: h,
    tags: N,
    maxVisibleTags: _,
    onBackspace: w,
    onSubmit: M
  }, j) => {
    if (t === "multiple" || t === "multiple-flex")
      return /* @__PURE__ */ l(
        z,
        {
          type: t,
          ref: j,
          label: a,
          name: i,
          placeholder: s,
          tags: N,
          maxVisibleTags: _,
          searchValue: x,
          onSearchChange: e,
          onBackspace: w,
          onSubmit: M,
          helpText: r === "error" ? void 0 : c,
          errorText: d,
          validationState: r,
          isDisabled: u,
          isOpen: f,
          leadingIcon: v,
          onFocus: m,
          onBlur: o,
          className: h,
          children: p
        }
      );
    const R = (T) => {
      e == null || e(T.value);
    };
    return /* @__PURE__ */ g("div", { className: E("fds-autocomplete", h), children: [
      /* @__PURE__ */ l(
        q,
        {
          ref: j,
          label: a,
          name: i,
          placeholder: s,
          value: x,
          onChange: R,
          onFocus: () => m == null ? void 0 : m(),
          onBlur: () => o == null ? void 0 : o(),
          helpText: r === "error" ? void 0 : c,
          errorText: d,
          validationState: r,
          isDisabled: u,
          icon: v,
          prefix: A,
          suffix: void 0,
          autoComplete: "off",
          "aria-expanded": f,
          "aria-haspopup": "listbox",
          "aria-autocomplete": "list"
        }
      ),
      f && p && /* @__PURE__ */ l("div", { className: "fds-autocomplete__popover", children: p })
    ] });
  }
);
G.displayName = "AutocompleteInput";
export {
  G as AutocompleteInput
};
