// @faclon-labs/design-sdk/dist/components/surfaces/Card/CardFooter.js
import { jsxs as s, jsx as o } from "react/jsx-runtime";
/* empty css               */
function _({
  title: e,
  subtitle: a,
  primaryAction: d,
  secondaryAction: r,
  showDivider: c = !0,
  layout: t = "Desktop",
  className: f
}) {
  const l = e || a, n = d || r, i = [
    "fds-card-footer",
    `fds-card-footer--${t.toLowerCase()}`,
    f
  ].filter(Boolean).join(" ");
  return /* @__PURE__ */ s("div", { className: i, children: [
    c && /* @__PURE__ */ o("hr", { className: "fds-card-footer__divider" }),
    /* @__PURE__ */ s("div", { className: "fds-card-footer__container", children: [
      l && /* @__PURE__ */ s("div", { className: "fds-card-footer__content", children: [
        e && /* @__PURE__ */ o("span", { className: "fds-card-footer__title BodyMediumSemibold", children: e }),
        a && /* @__PURE__ */ o("span", { className: "fds-card-footer__subtitle BodySmallRegular", children: a })
      ] }),
      n && /* @__PURE__ */ s("div", { className: "fds-card-footer__actions", children: [
        r && /* @__PURE__ */ o("div", { className: "fds-card-footer__action", children: r }),
        d && /* @__PURE__ */ o("div", { className: "fds-card-footer__action", children: d })
      ] })
    ] })
  ] });
}
_.displayName = "CardFooter";
export {
  _ as CardFooter
};
