// @faclon-labs/design-sdk/dist/components/surfaces/Card/CardHeader.js
import { jsxs as e, jsx as d } from "react/jsx-runtime";
/* empty css               */
function o({
  title: s,
  subtitle: a,
  leadingItem: c,
  trailingItem: r,
  counter: l,
  showDivider: i = !0,
  className: n
}) {
  const h = ["fds-card-header", n].filter(Boolean).join(" ");
  return /* @__PURE__ */ e("div", { className: h, children: [
    /* @__PURE__ */ e("div", { className: "fds-card-header__container", children: [
      /* @__PURE__ */ e("div", { className: "fds-card-header__left", children: [
        c,
        /* @__PURE__ */ e("div", { className: "fds-card-header__content", children: [
          /* @__PURE__ */ e("div", { className: "fds-card-header__title-container", children: [
            /* @__PURE__ */ d("span", { className: "fds-card-header__title BodyLargeSemibold", children: s }),
            l
          ] }),
          a && /* @__PURE__ */ d("span", { className: "fds-card-header__subtitle BodySmallRegular", children: a })
        ] })
      ] }),
      r && /* @__PURE__ */ d("div", { className: "fds-card-header__right", children: r })
    ] }),
    i && /* @__PURE__ */ d("hr", { className: "fds-card-header__divider" })
  ] });
}
o.displayName = "CardHeader";
export {
  o as CardHeader
};
