// @faclon-labs/design-sdk/dist/components/product/ProductAccordion/ProductAccordionItem.js
import { jsxs as t, jsx as e } from "react/jsx-runtime";
import { X as v, ChevronDown as y } from "react-feather";
import { cn as c } from "../../../utils/cn.js";
/* empty css                         */
function b({
  title: n,
  subtitle: m,
  isExpanded: s = !1,
  isActive: a = !0,
  isDisabled: d = !1,
  leadingItem: l,
  trailingIcon: r,
  onToggle: o,
  onClose: i,
  children: p,
  className: _,
  ...f
}) {
  const h = s && a && !d, N = (u) => {
    u.stopPropagation(), i == null || i();
  };
  return /* @__PURE__ */ t(
    "div",
    {
      className: c(
        "fds-pa-item",
        s && a && "fds-pa-item--expanded",
        !a && "fds-pa-item--inactive",
        d && "fds-pa-item--disabled",
        _
      ),
      ...f,
      children: [
        /* @__PURE__ */ t(
          "button",
          {
            type: "button",
            className: "fds-pa-item__header",
            onClick: d ? void 0 : o,
            disabled: d,
            "aria-expanded": s && a,
            children: [
              /* @__PURE__ */ t("div", { className: "fds-pa-item__title-area", children: [
                l,
                /* @__PURE__ */ t("div", { className: "fds-pa-item__title-section", children: [
                  /* @__PURE__ */ t("div", { className: "fds-pa-item__heading-row", children: [
                    /* @__PURE__ */ e("span", { className: "fds-pa-item__title BodyMediumMedium", children: n }),
                    r && /* @__PURE__ */ e("span", { className: "fds-pa-item__trailing-icon", children: r })
                  ] }),
                  m && /* @__PURE__ */ e("span", { className: "fds-pa-item__subtitle BodySmallRegular", children: m })
                ] })
              ] }),
              i && /* @__PURE__ */ e(
                "span",
                {
                  className: "fds-pa-item__close",
                  role: "button",
                  tabIndex: d ? -1 : 0,
                  onClick: N,
                  "aria-label": "Close",
                  children: /* @__PURE__ */ e(v, { size: 16 })
                }
              ),
              a && /* @__PURE__ */ e("span", { className: c("fds-pa-item__chevron", s && "fds-pa-item__chevron--open"), children: /* @__PURE__ */ e(y, { size: 16 }) })
            ]
          }
        ),
        h && /* @__PURE__ */ e("div", { className: "fds-pa-item__body", children: p })
      ]
    }
  );
}
b.displayName = "ProductAccordionItem";
export {
  b as ProductAccordionItem
};
