// @faclon-labs/design-sdk/dist/components/inputs/MultiSelectInput/MultiSelectInput.js
import { jsx as e, jsxs as l, Fragment as H } from "react/jsx-runtime";
import { forwardRef as U, useId as W, useRef as X, useCallback as y } from "react";
import { ChevronDown as T } from "react-feather";
import { cn as a } from "../../../utils/cn.js";
import { Tag as q } from "../../data-display/Tag/Tag.js";
import { InputFieldHeader as Y } from "../../forms/InputFieldHeader/InputFieldHeader.js";
import { InputFieldFooter as Z } from "../../forms/InputFieldFooter/InputFieldFooter.js";
/* empty css                     */
const $ = U(
  ({
    type: A = "multiple",
    label: h,
    name: w,
    placeholder: R = "Select",
    tags: d = [],
    maxVisibleTags: g = 2,
    helpText: B,
    errorText: C,
    validationState: G = "none",
    isDisabled: s = !1,
    isOpen: r = !1,
    searchValue: i,
    onSearchChange: o,
    onBackspace: n,
    onSubmit: m,
    leadingIcon: c,
    leadingText: f,
    children: M,
    onClick: p,
    onFocus: F,
    onBlur: b,
    className: J
  }, L) => {
    const N = W(), P = X(null), u = L ?? P, _ = A === "multiple-flex", x = d.length > 0, v = G === "error", z = v ? C : B, I = _ ? d : d.slice(0, g), K = _ ? 0 : d.length - g, Q = y(
      (t) => {
        var E;
        s || ((E = u.current) == null || E.focus(), p == null || p(t));
      },
      [s, p, u]
    ), j = y(
      (t) => {
        o == null || o(t.target.value);
      },
      [o]
    ), D = y(
      (t) => {
        t.key === "Backspace" && !i && (n == null || n()), t.key === "Enter" && i && (t.preventDefault(), m == null || m(i));
      },
      [i, n, m]
    );
    return /* @__PURE__ */ e("div", { className: a("fds-multi-select__root", J), children: /* @__PURE__ */ l(
      "div",
      {
        className: a(
          "fds-text-input",
          s && "fds-text-input--disabled",
          v && "fds-text-input--error",
          r && "fds-multi-select--open"
        ),
        children: [
          /* @__PURE__ */ e(Y, { label: h, htmlFor: N }),
          /* @__PURE__ */ l("div", { className: "fds-text-input__field-wrapper fds-multi-select__field-wrapper", children: [
            /* @__PURE__ */ e(
              "div",
              {
                className: a(
                  "fds-text-input__field fds-multi-select__trigger",
                  _ && "fds-multi-select__trigger--flex"
                ),
                role: "combobox",
                "aria-expanded": r,
                "aria-haspopup": "listbox",
                "aria-disabled": s || void 0,
                onClick: Q,
                children: _ ? (
                  /* ── Multiple Flex: column layout ───────────────────── */
                  /* @__PURE__ */ l(H, { children: [
                    /* @__PURE__ */ l("span", { className: "fds-multi-select__leading-col", children: [
                      x && /* @__PURE__ */ e("span", { className: "fds-multi-select__tags fds-multi-select__tags--flex", children: I.map((t) => /* @__PURE__ */ e(
                        q,
                        {
                          id: t.label,
                          label: t.label,
                          size: "Medium",
                          isDisabled: s,
                          onDismiss: t.onDismiss
                        }
                      )) }),
                      /* @__PURE__ */ l("span", { className: "fds-multi-select__input-row", children: [
                        c && /* @__PURE__ */ e("span", { className: "fds-text-input__icon", children: c }),
                        f && /* @__PURE__ */ e("span", { className: "fds-text-input__prefix BodyMediumRegular", children: f }),
                        /* @__PURE__ */ e(
                          "input",
                          {
                            ref: u,
                            id: N,
                            type: "text",
                            role: "searchbox",
                            className: "fds-multi-select__input BodyMediumRegular",
                            name: w,
                            placeholder: x ? "" : R,
                            value: i ?? "",
                            disabled: s,
                            "aria-label": h,
                            "aria-autocomplete": "list",
                            autoComplete: "off",
                            onChange: j,
                            onKeyDown: D,
                            onFocus: F,
                            onBlur: b
                          }
                        )
                      ] })
                    ] }),
                    /* @__PURE__ */ e("span", { className: "fds-multi-select__trailing--flex", children: /* @__PURE__ */ e(
                      "span",
                      {
                        className: a(
                          "fds-multi-select__chevron",
                          r && "fds-multi-select__chevron--open"
                        ),
                        children: /* @__PURE__ */ e(T, { size: 16 })
                      }
                    ) })
                  ] })
                ) : (
                  /* ── Multiple: single-row layout ────────────────────── */
                  /* @__PURE__ */ l(H, { children: [
                    /* @__PURE__ */ l("span", { className: "fds-text-input__leading", children: [
                      c && /* @__PURE__ */ e("span", { className: "fds-text-input__icon", children: c }),
                      f && /* @__PURE__ */ e("span", { className: "fds-text-input__prefix BodyMediumRegular", children: f }),
                      x && /* @__PURE__ */ l("span", { className: "fds-multi-select__tags", children: [
                        I.map((t) => /* @__PURE__ */ e(
                          q,
                          {
                            id: t.label,
                            label: t.label,
                            size: "Medium",
                            isDisabled: s,
                            onDismiss: t.onDismiss
                          }
                        )),
                        K > 0 && /* @__PURE__ */ l("span", { className: "fds-multi-select__overflow BodySmallRegular", children: [
                          "+",
                          K,
                          " more"
                        ] })
                      ] }),
                      /* @__PURE__ */ e(
                        "input",
                        {
                          ref: u,
                          id: N,
                          type: "text",
                          role: "searchbox",
                          className: "fds-multi-select__input BodyMediumRegular",
                          name: w,
                          placeholder: x ? "" : R,
                          value: i ?? "",
                          disabled: s,
                          "aria-label": h,
                          "aria-autocomplete": "list",
                          autoComplete: "off",
                          onChange: j,
                          onKeyDown: D,
                          onFocus: F,
                          onBlur: b
                        }
                      )
                    ] }),
                    /* @__PURE__ */ e("span", { className: "fds-text-input__trailing", children: /* @__PURE__ */ e(
                      "span",
                      {
                        className: a(
                          "fds-multi-select__chevron",
                          r && "fds-multi-select__chevron--open"
                        ),
                        children: /* @__PURE__ */ e(T, { size: 16 })
                      }
                    ) })
                  ] })
                )
              }
            ),
            r && M && /* @__PURE__ */ e("div", { className: "fds-multi-select__popover", children: M })
          ] }),
          z && /* @__PURE__ */ e(
            Z,
            {
              helpText: z,
              state: v ? "error" : "default"
            }
          )
        ]
      }
    ) });
  }
);
$.displayName = "MultiSelectInput";
export {
  $ as MultiSelectInput
};
