// @faclon-labs/design-sdk/dist/components/product/ColorPicker/ColorPicker.js
import { jsxs as t, jsx as e } from "react/jsx-runtime";
import { useState as _, useCallback as u } from "react";
import { cn as s } from "../../../utils/cn.js";
/* empty css                */
function h() {
  return /* @__PURE__ */ e("svg", { width: "24", height: "24", viewBox: "0 0 24 24", fill: "none", xmlns: "http://www.w3.org/2000/svg", "aria-hidden": "true", children: /* @__PURE__ */ e("path", { d: "M12 4.81V19c-3.31 0-6-2.63-6-5.87 0-1.56.62-3.03 1.75-4.14L12 4.81zM12 2L6.35 7.56C4.9 8.99 4 10.96 4 13.13 4 17.48 7.58 21 12 21s8-3.52 8-7.87c0-2.17-.9-4.14-2.35-5.57L12 2z", fill: "currentColor" }) });
}
function b({
  activeTab: i,
  onTabChange: r,
  presetsPanel: n,
  pickerPanel: p,
  onThemeToggle: l,
  isDark: v = !1,
  className: d,
  ...k
}) {
  const [f, m] = _("presets"), c = i ?? f, o = u(
    (a) => {
      i || m(a), r == null || r(a);
    },
    [i, r]
  );
  return /* @__PURE__ */ t("div", { className: s("fds-color-picker", d), ...k, children: [
    /* @__PURE__ */ t("div", { className: "fds-color-picker__header", children: [
      /* @__PURE__ */ t("div", { className: "fds-color-picker__tabs", children: [
        /* @__PURE__ */ e(
          "button",
          {
            type: "button",
            className: s("fds-color-picker__tab", c === "presets" && "fds-color-picker__tab--active"),
            onClick: () => o("presets"),
            children: "PRESETS"
          }
        ),
        /* @__PURE__ */ e(
          "button",
          {
            type: "button",
            className: s("fds-color-picker__tab", c === "picker" && "fds-color-picker__tab--active"),
            onClick: () => o("picker"),
            children: "COLOR PICKER"
          }
        )
      ] }),
      l && /* @__PURE__ */ e(
        "button",
        {
          type: "button",
          className: "fds-color-picker__theme-toggle",
          onClick: l,
          "aria-label": "Toggle theme",
          children: /* @__PURE__ */ e(h, {})
        }
      )
    ] }),
    /* @__PURE__ */ t("div", { className: "fds-color-picker__panel", children: [
      c === "presets" && n,
      c === "picker" && p
    ] })
  ] });
}
b.displayName = "ColorPicker";
export {
  b as ColorPicker
};
