// @faclon-labs/design-sdk/dist/components/product/ColorPicker/ColorPickerPresets.js
import { jsx as e, jsxs as b } from "react/jsx-runtime";
import { cn as r } from "../../../utils/cn.js";
/* empty css                       */
const m = [
  { name: "BLUE", colors: ["#e9f2ff", "#cce0ff", "#85b8ff", "#579dff", "#388bff", "#1d7afc", "#0c66e4", "#0055cc", "#09326c", "#092957"] },
  { name: "MAGENTA", colors: ["#f3f0ff", "#dfd8fd", "#b8acf6", "#9f8fef", "#8f7ee7", "#8270db", "#6e5dc6", "#5e4db2", "#352c63", "#2b2451"] },
  { name: "GREEN", colors: ["#dffcf0", "#baf3db", "#7ee2b8", "#4bce97", "#2abb7f", "#22a06b", "#1f845a", "#216e4e", "#164b35", "#143c2b"] },
  { name: "RED", colors: ["#ffedeb", "#ffd2cc", "#ff9c8f", "#f87462", "#ef5c48", "#e34935", "#ca3521", "#ae2a19", "#601e16", "#4f1c16"] },
  { name: "ORANGE", colors: ["#fff4e5", "#ffe2bd", "#fec57b", "#faa53d", "#f18d13", "#d97008", "#b65c02", "#974f0c", "#5f3811", "#4a2b0f"] },
  { name: "NEUTRALS", colors: ["#dee4ea", "#c7d1db", "#9fadbc", "#8c9bab", "#596773", "#454f59", "#2c333a", "#22272b", "#1d2125", "#101214"] }
];
function o({
  palettes: d = m,
  selectedColor: s,
  onColorSelect: a
}) {
  return /* @__PURE__ */ e("div", { className: "fds-color-presets", children: d.map((c) => /* @__PURE__ */ b("div", { id: c.name, className: "fds-color-presets__palette", children: [
    /* @__PURE__ */ e("span", { className: "fds-color-presets__label", children: c.name }),
    /* @__PURE__ */ e("div", { className: "fds-color-presets__swatches", children: c.colors.map((f) => /* @__PURE__ */ e(
      "button",
      {
        id: f,
        type: "button",
        className: r(
          "fds-color-presets__swatch",
          (s == null ? void 0 : s.toLowerCase()) === f.toLowerCase() && "fds-color-presets__swatch--selected"
        ),
        style: { backgroundColor: f },
        onClick: () => a == null ? void 0 : a(f),
        "aria-label": f
      }
    )) })
  ] })) });
}
o.displayName = "ColorPickerPresets";
export {
  o as ColorPickerPresets,
  m as DEFAULT_PALETTES
};
