// @faclon-labs/design-sdk/dist/components/product/ColorPicker/ColorPickerPanel.js
import { jsxs as l, jsx as n } from "react/jsx-runtime";
import { useRef as x, useCallback as D, useState as h, useEffect as L } from "react";
/* empty css                     */
function V(v) {
  return `hsl(${v}, 100%, 50%)`;
}
function W({
  hue: v = 30,
  saturation: j = 80,
  brightness: F = 90,
  opacity: Y = 100,
  r: d = 232,
  g: u = 232,
  b: m = 232,
  hex: y = "#FF0011",
  onHueChange: _,
  onSaturationBrightnessChange: f,
  onOpacityChange: M,
  onRgbChange: E,
  onHexChange: w
}) {
  const $ = x(null), k = x(null), b = x(null), T = D(
    (e) => {
      var r;
      const t = (r = $.current) == null ? void 0 : r.getBoundingClientRect();
      if (!t) return;
      const s = (o, p) => {
        const J = Math.max(0, Math.min(100, (o - t.left) / t.width * 100)), Q = Math.max(0, Math.min(100, 100 - (p - t.top) / t.height * 100));
        f == null || f(J, Q);
      };
      s(e.clientX, e.clientY);
      const c = (o) => s(o.clientX, o.clientY), a = () => {
        document.removeEventListener("mousemove", c), document.removeEventListener("mouseup", a);
      };
      document.addEventListener("mousemove", c), document.addEventListener("mouseup", a);
    },
    [f]
  ), q = D(
    (e) => {
      var r;
      const t = (r = k.current) == null ? void 0 : r.getBoundingClientRect();
      if (!t) return;
      const s = (o) => {
        const p = Math.max(0, Math.min(360, (o - t.left) / t.width * 360));
        _ == null || _(Math.round(p));
      };
      s(e.clientX);
      const c = (o) => s(o.clientX), a = () => {
        document.removeEventListener("mousemove", c), document.removeEventListener("mouseup", a);
      };
      document.addEventListener("mousemove", c), document.addEventListener("mouseup", a);
    },
    [_]
  ), z = D(
    (e) => {
      var r;
      const t = (r = b.current) == null ? void 0 : r.getBoundingClientRect();
      if (!t) return;
      const s = (o) => {
        const p = Math.max(0, Math.min(100, (o - t.left) / t.width * 100));
        M == null || M(Math.round(p));
      };
      s(e.clientX);
      const c = (o) => s(o.clientX), a = () => {
        document.removeEventListener("mousemove", c), document.removeEventListener("mouseup", a);
      };
      document.addEventListener("mousemove", c), document.addEventListener("mouseup", a);
    },
    [M]
  ), A = V(v), [R, X] = h(String(d)), [B, K] = h(String(u)), [P, S] = h(String(m)), [G, I] = h(y);
  L(() => {
    X(String(d));
  }, [d]), L(() => {
    K(String(u));
  }, [u]), L(() => {
    S(String(m));
  }, [m]), L(() => {
    I(y);
  }, [y]);
  const i = () => {
    E == null || E(
      Math.max(0, Math.min(255, parseInt(R) || 0)),
      Math.max(0, Math.min(255, parseInt(B) || 0)),
      Math.max(0, Math.min(255, parseInt(P) || 0))
    );
  }, U = () => {
    w == null || w(G);
  }, N = (e) => (t) => {
    t.key === "Enter" && e();
  };
  return /* @__PURE__ */ l("div", { className: "fds-color-panel", children: [
    /* @__PURE__ */ l(
      "div",
      {
        ref: $,
        className: "fds-color-panel__canvas",
        style: { background: A },
        onMouseDown: T,
        children: [
          /* @__PURE__ */ n("div", { className: "fds-color-panel__canvas-saturation" }),
          /* @__PURE__ */ n("div", { className: "fds-color-panel__canvas-brightness" }),
          /* @__PURE__ */ n(
            "div",
            {
              className: "fds-color-panel__picker",
              style: { left: `${j}%`, top: `${100 - F}%` }
            }
          )
        ]
      }
    ),
    /* @__PURE__ */ l("div", { ref: k, className: "fds-color-panel__hue", onMouseDown: q, children: [
      /* @__PURE__ */ n("div", { className: "fds-color-panel__hue-track" }),
      /* @__PURE__ */ n("div", { className: "fds-color-panel__slider-thumb", style: { left: `${v / 360 * 100}%` } })
    ] }),
    /* @__PURE__ */ l("div", { ref: b, className: "fds-color-panel__opacity", onMouseDown: z, children: [
      /* @__PURE__ */ n("div", { className: "fds-color-panel__opacity-track", style: { background: `linear-gradient(to right, rgba(${d},${u},${m},0), rgb(${d},${u},${m}))` } }),
      /* @__PURE__ */ n("div", { className: "fds-color-panel__slider-thumb", style: { left: `${Y}%` } })
    ] }),
    /* @__PURE__ */ l("div", { className: "fds-color-panel__inputs", children: [
      /* @__PURE__ */ l("div", { className: "fds-color-panel__rgb", children: [
        /* @__PURE__ */ l("div", { className: "fds-color-panel__input-group", children: [
          /* @__PURE__ */ n(
            "input",
            {
              type: "text",
              inputMode: "numeric",
              className: "fds-color-panel__input",
              value: R,
              onChange: (e) => X(e.target.value),
              onBlur: i,
              onKeyDown: N(i)
            }
          ),
          /* @__PURE__ */ n("span", { className: "fds-color-panel__input-label", children: "R" })
        ] }),
        /* @__PURE__ */ l("div", { className: "fds-color-panel__input-group", children: [
          /* @__PURE__ */ n(
            "input",
            {
              type: "text",
              inputMode: "numeric",
              className: "fds-color-panel__input",
              value: B,
              onChange: (e) => K(e.target.value),
              onBlur: i,
              onKeyDown: N(i)
            }
          ),
          /* @__PURE__ */ n("span", { className: "fds-color-panel__input-label", children: "G" })
        ] }),
        /* @__PURE__ */ l("div", { className: "fds-color-panel__input-group", children: [
          /* @__PURE__ */ n(
            "input",
            {
              type: "text",
              inputMode: "numeric",
              className: "fds-color-panel__input",
              value: P,
              onChange: (e) => S(e.target.value),
              onBlur: i,
              onKeyDown: N(i)
            }
          ),
          /* @__PURE__ */ n("span", { className: "fds-color-panel__input-label", children: "B" })
        ] })
      ] }),
      /* @__PURE__ */ l("div", { className: "fds-color-panel__input-group fds-color-panel__input-group--hex", children: [
        /* @__PURE__ */ n(
          "input",
          {
            type: "text",
            className: "fds-color-panel__input",
            value: G,
            onChange: (e) => I(e.target.value),
            onBlur: U,
            onKeyDown: N(U)
          }
        ),
        /* @__PURE__ */ n("span", { className: "fds-color-panel__input-label", children: "HEX" })
      ] })
    ] })
  ] });
}
W.displayName = "ColorPickerPanel";
export {
  W as ColorPickerPanel
};
