// @faclon-labs/design-sdk/dist/components/actions/Button/Button.js
import { jsx as s, jsxs as y, Fragment as P } from "react/jsx-runtime";
import { forwardRef as B } from "react";
import { Spinner as C } from "../../feedback/Spinner/Spinner.js";
/* empty css           */
const v = B(
  ({
    variant: l = "Primary",
    color: m = "Primary",
    size: o = "Medium",
    label: r,
    leadingIcon: n,
    trailingIcon: d,
    iconOnly: a = !1,
    isFullWidth: f = !1,
    isDisabled: i = !1,
    isLoading: e = !1,
    className: b,
    children: t,
    ..._
  }, p) => {
    const c = {
      Large: "BodyLargeSemibold",
      Medium: "BodyMediumSemibold",
      Small: "BodySmallSemibold",
      XSmall: "BodySmallSemibold"
    }, N = [
      "fds-btn",
      `fds-btn--variant-${l.toLowerCase()}`,
      `fds-btn--color-${m.toLowerCase()}`,
      `fds-btn--size-${o.toLowerCase()}`,
      c[o],
      a && "fds-btn--icon-only",
      f && "fds-btn--full-width",
      e && "fds-btn--loading",
      b
    ].filter(Boolean).join(" "), S = o === "Large" ? "Large" : "Medium", u = l === "Primary" ? "White" : {
      Primary: "Brand",
      Negative: "Negative",
      Positive: "Positive"
    }[m];
    return /* @__PURE__ */ s(
      "button",
      {
        ref: p,
        className: N,
        type: "button",
        disabled: i || e,
        "aria-disabled": i || e || void 0,
        "aria-busy": e || void 0,
        ..._,
        children: e ? /* @__PURE__ */ s(C, { color: u, size: S }) : /* @__PURE__ */ y(P, { children: [
          n && /* @__PURE__ */ s("span", { className: "fds-btn__icon fds-btn__icon--leading", children: n }),
          !a && r && /* @__PURE__ */ s("span", { className: "fds-btn__label", children: r }),
          !a && !r && t && /* @__PURE__ */ s("span", { className: "fds-btn__label", children: t }),
          a && !n && !d && t && /* @__PURE__ */ s("span", { className: "fds-btn__icon", children: t }),
          d && /* @__PURE__ */ s("span", { className: "fds-btn__icon fds-btn__icon--trailing", children: d })
        ] })
      }
    );
  }
);
v.displayName = "Button";
export {
  v as Button
};
