// @faclon-labs/design-sdk/dist/index.js
/* empty css                  */
/* empty css                                     */
/* empty css                                             */
/* empty css                                                     */
/* empty css                                                      */
/* empty css                                        */
/* empty css                                    */
/* empty css                                                */
/* empty css                                        */
/* empty css                                        */
/* empty css                                             */
/* empty css                                  */
/* empty css                                       */
/* empty css                                        */
/* empty css                                          */
/* empty css                                              */
/* empty css                                                        */
/* empty css                                                          */
/* empty css                                    */
/* empty css                                          */
/* empty css                                                       */
/* empty css                                                       */
/* empty css                                      */
/* empty css                                    */
/* empty css                                       */
/* empty css                                               */
/* empty css                                                */
/* empty css                                  */
/* empty css                                        */
/* empty css                                        */
/* empty css                                      */
/* empty css                                             */
/* empty css                                              */
/* empty css                                                */
/* empty css                                                       */
/* empty css                                                  */
/* empty css                                                          */
/* empty css                                                    */
/* empty css                                                         */
/* empty css                                                    */
/* empty css                                                    */
/* empty css                                                         */
/* empty css                                    */
/* empty css                                          */
/* empty css                                        */
/* empty css                                          */
/* empty css                                               */
/* empty css                                                */
/* empty css                                               */
/* empty css                                                    */
/* empty css                                                      */
/* empty css                                                         */
/* empty css                                                         */
/* empty css                                             */
/* empty css                                                    */
/* empty css                                                    */
/* empty css                                               */
/* empty css                                                 */
/* empty css                                                   */
/* empty css                                                  */
/* empty css                                                 */
/* empty css                                                 */
/* empty css                                                    */
/* empty css                                                */
/* empty css                                                      */
/* empty css                                                       */
/* empty css                                         */
/* empty css                                                    */
/* empty css                                                     */
/* empty css                                                             */
/* empty css                                                      */
/* empty css                                                       */
import { useClickOutside as Lo } from "./hooks/useClickOutside.js";
import { useMediaQuery as bo } from "./hooks/useMediaQuery.js";
import { useKeyboard as Ao } from "./hooks/useKeyboard.js";
import { cn as Mo } from "./utils/cn.js";
import { Accordion as ho } from "./components/surfaces/Accordion/Accordion.js";
import { AccordionItem as Fo } from "./components/surfaces/Accordion/AccordionItem.js";
import { AccordionLeadingItem as yo } from "./components/surfaces/Accordion/AccordionLeadingItem.js";
import { ActionListItem as Ro } from "./components/overlays/DropdownMenu/ActionListItem.js";
import { ActionListItemGroup as vo } from "./components/overlays/DropdownMenu/ActionListItemGroup.js";
import { AutocompleteInput as Yo } from "./components/inputs/AutocompleteInput/AutocompleteInput.js";
import { Badge as Oo } from "./components/data-display/Badge/Badge.js";
import { Button as Wo } from "./components/actions/Button/Button.js";
import { CalendarBase as jo } from "./components/product/DatePicker/CalendarBase.js";
import { CalendarDayCell as zo } from "./components/product/DatePicker/CalendarDayCell.js";
import { CalendarFooter as No } from "./components/product/DatePicker/CalendarFooter.js";
import { CalendarHeader as Xo } from "./components/product/DatePicker/CalendarHeader.js";
import { CalendarWeekdays as $o } from "./components/product/DatePicker/CalendarWeekdays.js";
import { Card as rr } from "./components/surfaces/Card/Card.js";
import { CardBody as er } from "./components/surfaces/Card/CardBody.js";
import { CardFooter as pr } from "./components/surfaces/Card/CardFooter.js";
import { CardHeader as ar } from "./components/surfaces/Card/CardHeader.js";
import { CardLeadingItem as fr } from "./components/surfaces/Card/CardLeadingItem.js";
import { CardTrailingItem as nr } from "./components/surfaces/Card/CardTrailingItem.js";
import { Checkbox as cr } from "./components/inputs/Checkbox/Checkbox.js";
import { CheckboxGroup as sr } from "./components/inputs/Checkbox/CheckboxGroup.js";
import { ColorPicker as Cr } from "./components/product/ColorPicker/ColorPicker.js";
import { ColorPickerPanel as Tr } from "./components/product/ColorPicker/ColorPickerPanel.js";
import { ColorPickerPresets as Lr, DEFAULT_PALETTES as Pr } from "./components/product/ColorPicker/ColorPickerPresets.js";
import { ComparisonButton as Br } from "./components/product/ComparisonToggle/ComparisonButton.js";
import { ComparisonToggle as kr } from "./components/product/ComparisonToggle/ComparisonToggle.js";
import { DatePicker as Sr } from "./components/product/DatePicker/DatePicker.js";
import { DatePickerPopover as wr } from "./components/product/DatePicker/DatePickerPopover.js";
import { DatePickerTrigger as Hr } from "./components/product/DatePicker/DatePickerTrigger.js";
import { DatePresetBase as Gr } from "./components/product/DatePicker/DatePresetBase.js";
import { DatePresetSidebar as Er } from "./components/product/DatePicker/DatePresetSidebar.js";
import { DateSelectorButton as Ur } from "./components/product/DateSelector/DateSelectorButton.js";
import { DateSelectorGroup as Kr } from "./components/product/DateSelector/DateSelectorGroup.js";
import { Divider as Qr } from "./components/layout/Divider/Divider.js";
import { DropdownFooter as _r } from "./components/overlays/DropdownMenu/DropdownFooter.js";
import { DropdownHeader as qr } from "./components/overlays/DropdownMenu/DropdownHeader.js";
import { DropdownLeadingItem as Jr } from "./components/overlays/DropdownMenu/DropdownLeadingItem.js";
import { DropdownMenu as Vr } from "./components/overlays/DropdownMenu/DropdownMenu.js";
import { DropdownTrailingItem as Zr } from "./components/overlays/DropdownMenu/DropdownTrailingItem.js";
import { IconButton as ot } from "./components/actions/IconButton/IconButton.js";
import { Indicator as tt } from "./components/data-display/Indicator/Indicator.js";
import { InputFieldFooter as mt } from "./components/forms/InputFieldFooter/InputFieldFooter.js";
import { InputFieldHeader as it } from "./components/forms/InputFieldHeader/InputFieldHeader.js";
import { LinkButton as xt } from "./components/actions/LinkButton/LinkButton.js";
import { ListCard as dt } from "./components/product/ListCard/ListCard.js";
import { ListCardLeadingItem as lt } from "./components/product/ListCard/ListCardLeadingItem.js";
import { ListCardTrailingItem as gt } from "./components/product/ListCard/ListCardTrailingItem.js";
import { Modal as ut } from "./components/overlays/Modal/Modal.js";
import { ModalBody as It } from "./components/overlays/Modal/ModalBody.js";
import { ModalFooter as Dt } from "./components/overlays/Modal/ModalFooter.js";
import { ModalHeader as Pt } from "./components/overlays/Modal/ModalHeader.js";
import { ModalLeadingItem as Bt } from "./components/overlays/Modal/ModalLeadingItem.js";
import { ModalTrailingItem as kt } from "./components/overlays/Modal/ModalTrailingItem.js";
import { MonthYearCell as St } from "./components/product/DatePicker/MonthYearCell.js";
import { MultiSelectInput as wt } from "./components/inputs/MultiSelectInput/MultiSelectInput.js";
import { PALeadingItem as Ht } from "./components/product/ProductAccordion/PALeadingItem.js";
import { PATrailingItem as Gt } from "./components/product/ProductAccordion/PATrailingItem.js";
import { ProductAccordionItem as Et } from "./components/product/ProductAccordion/ProductAccordionItem.js";
import { Radio as Ut } from "./components/inputs/Radio/Radio.js";
import { RadioGroup as Kt } from "./components/inputs/Radio/RadioGroup.js";
import { SelectInput as Qt } from "./components/inputs/SelectInput/SelectInput.js";
import { Spinner as _t } from "./components/feedback/Spinner/Spinner.js";
import { Switch as qt } from "./components/inputs/Switch/Switch.js";
import { SwitchButtonBase as Jt } from "./components/actions/SwitchButton/SwitchButtonBase.js";
import { SwitchButtonGroup as Vt } from "./components/actions/SwitchButton/SwitchButtonGroup.js";
import { TabItem as Zt } from "./components/navigation/Tabs/TabItem.js";
import { Tabs as oe } from "./components/navigation/Tabs/Tabs.js";
import { TabsLeadingItem as te } from "./components/navigation/Tabs/TabsLeadingItem.js";
import { TabsTrailingItem as me } from "./components/navigation/Tabs/TabsTrailingItem.js";
import { Tag as ie } from "./components/data-display/Tag/Tag.js";
import { TextArea as xe } from "./components/inputs/TextArea/TextArea.js";
import { TextInput as de } from "./components/inputs/TextInput/TextInput.js";
import { UploadCta as le } from "./components/inputs/UploadCta/UploadCta.js";
import { formatDate as ge, formatTime as se, generateCalendarDays as ue, generateMonths as Ce, generateYears as Ie, getHeaderLabel as Te, getPresetDateRange as De } from "./components/product/DatePicker/datePickerUtils.js";
import { hexToRgb as Pe, hsbToRgb as be, rgbToHex as Be, rgbToHsb as Ae } from "./components/product/ColorPicker/colorUtils.js";
export {
  ho as Accordion,
  Fo as AccordionItem,
  yo as AccordionLeadingItem,
  Ro as ActionListItem,
  vo as ActionListItemGroup,
  Yo as AutocompleteInput,
  Oo as Badge,
  Wo as Button,
  jo as CalendarBase,
  zo as CalendarDayCell,
  No as CalendarFooter,
  Xo as CalendarHeader,
  $o as CalendarWeekdays,
  rr as Card,
  er as CardBody,
  pr as CardFooter,
  ar as CardHeader,
  fr as CardLeadingItem,
  nr as CardTrailingItem,
  cr as Checkbox,
  sr as CheckboxGroup,
  Cr as ColorPicker,
  Tr as ColorPickerPanel,
  Lr as ColorPickerPresets,
  Br as ComparisonButton,
  kr as ComparisonToggle,
  Pr as DEFAULT_PALETTES,
  Sr as DatePicker,
  wr as DatePickerPopover,
  Hr as DatePickerTrigger,
  Gr as DatePresetBase,
  Er as DatePresetSidebar,
  Ur as DateSelectorButton,
  Kr as DateSelectorGroup,
  Qr as Divider,
  _r as DropdownFooter,
  qr as DropdownHeader,
  Jr as DropdownLeadingItem,
  Vr as DropdownMenu,
  Zr as DropdownTrailingItem,
  ot as IconButton,
  tt as Indicator,
  mt as InputFieldFooter,
  it as InputFieldHeader,
  xt as LinkButton,
  dt as ListCard,
  lt as ListCardLeadingItem,
  gt as ListCardTrailingItem,
  ut as Modal,
  It as ModalBody,
  Dt as ModalFooter,
  Pt as ModalHeader,
  Bt as ModalLeadingItem,
  kt as ModalTrailingItem,
  St as MonthYearCell,
  wt as MultiSelectInput,
  Ht as PALeadingItem,
  Gt as PATrailingItem,
  Et as ProductAccordionItem,
  Ut as Radio,
  Kt as RadioGroup,
  Qt as SelectInput,
  _t as Spinner,
  qt as Switch,
  Jt as SwitchButtonBase,
  Vt as SwitchButtonGroup,
  Zt as TabItem,
  oe as Tabs,
  te as TabsLeadingItem,
  me as TabsTrailingItem,
  ie as Tag,
  xe as TextArea,
  de as TextInput,
  le as UploadCta,
  Mo as cn,
  ge as formatDate,
  se as formatTime,
  ue as generateCalendarDays,
  Ce as generateMonths,
  Ie as generateYears,
  Te as getHeaderLabel,
  De as getPresetDateRange,
  Pe as hexToRgb,
  be as hsbToRgb,
  Be as rgbToHex,
  Ae as rgbToHsb,
  Lo as useClickOutside,
  Ao as useKeyboard,
  bo as useMediaQuery
};
