// @faclon-labs/design-sdk/dist/components/inputs/Switch/Switch.js
import { jsxs as i, Fragment as B, jsx as s } from "react/jsx-runtime";
import { forwardRef as L, useId as j, useCallback as y } from "react";
import { Divider as I } from "../../layout/Divider/Divider.js";
/* empty css           */
function M({ size: e }) {
  return /* @__PURE__ */ s(
    "svg",
    {
      width: e,
      height: e,
      viewBox: "0 0 12 12",
      fill: "none",
      xmlns: "http://www.w3.org/2000/svg",
      "aria-hidden": "true",
      className: "fds-switch__check-icon",
      children: /* @__PURE__ */ s(
        "path",
        {
          d: "M9.46967 2.46967C9.76256 2.17678 10.2373 2.17678 10.5302 2.46967C10.8231 2.76256 10.8231 3.23732 10.5302 3.53022L5.03022 9.03022C4.73732 9.32311 4.26256 9.32311 3.96967 9.03022L1.46967 6.53022C1.17678 6.23732 1.17678 5.76256 1.46967 5.46967C1.76256 5.17678 2.23732 5.17678 2.53022 5.46967L4.49994 7.4394L9.46967 2.46967Z",
          fill: "currentColor"
        }
      )
    }
  );
}
const F = {
  Small: 8,
  Medium: 10
}, D = L(
  ({
    size: e = "Medium",
    isChecked: t,
    defaultChecked: w,
    isDisabled: _ = !1,
    onChange: c,
    name: p,
    accessibilityLabel: u,
    label: d,
    helpText: r,
    trailingIcon: h,
    showDivider: N = !1,
    className: n,
    id: v,
    disabled: k,
    ...b
  }, C) => {
    const S = j(), l = v ?? S, a = p ?? "", o = _ || k || !1, g = y(
      (x) => {
        c == null || c({ name: a, checked: x.target.checked });
      },
      [a, c]
    ), f = [
      "fds-switch",
      `fds-switch--${e.toLowerCase()}`,
      o && "fds-switch--disabled",
      !d && n
    ].filter(Boolean).join(" "), m = /* @__PURE__ */ i(B, { children: [
      /* @__PURE__ */ s(
        "input",
        {
          ref: C,
          id: l,
          type: "checkbox",
          role: "switch",
          className: "fds-switch__input",
          name: a || void 0,
          checked: t,
          defaultChecked: w,
          disabled: o,
          "aria-label": u ?? d,
          "aria-checked": t,
          onChange: g,
          ...b
        }
      ),
      /* @__PURE__ */ s("span", { className: "fds-switch__wrapper", children: /* @__PURE__ */ s("span", { className: "fds-switch__track", children: /* @__PURE__ */ s("span", { className: "fds-switch__thumb", children: /* @__PURE__ */ s(M, { size: F[e] }) }) }) })
    ] });
    return d ? /* @__PURE__ */ i("div", { className: ["fds-switch-field", n].filter(Boolean).join(" "), children: [
      /* @__PURE__ */ s("label", { className: f, htmlFor: l, children: /* @__PURE__ */ i("div", { className: "fds-switch-field__row", children: [
        /* @__PURE__ */ i("div", { className: "fds-switch-field__text", children: [
          /* @__PURE__ */ i("div", { className: "fds-switch-field__heading", children: [
            /* @__PURE__ */ s("span", { className: "fds-switch-field__title BodySmallSemibold", children: d }),
            h && /* @__PURE__ */ s("span", { className: "fds-switch-field__icon", children: h })
          ] }),
          r && /* @__PURE__ */ s("span", { className: "fds-switch-field__help BodySmallRegular", children: r })
        ] }),
        m
      ] }) }),
      N && /* @__PURE__ */ s(I, { variant: "Muted" })
    ] }) : /* @__PURE__ */ s("label", { className: f, htmlFor: l, children: m });
  }
);
D.displayName = "Switch";
export {
  D as Switch
};
