// @faclon-labs/design-sdk/dist/components/inputs/UploadCta/UploadCta.js
import { jsxs as l, jsx as e } from "react/jsx-runtime";
import { useRef as y, useState as C, useCallback as n } from "react";
import { cn as N } from "../../../utils/cn.js";
/* empty css              */
function x({
  bodyText: p = "Drag files here or",
  linkText: u = "Upload",
  isDisabled: r = !1,
  accept: s,
  multiple: c = !1,
  onFilesSelect: t,
  className: g,
  ...h
}) {
  const f = y(null), [i, d] = C(!1), o = n(() => {
    var a;
    r || (a = f.current) == null || a.click();
  }, [r]), m = n(
    (a) => {
      a.target.files && a.target.files.length > 0 && (t == null || t(a.target.files)), a.target.value = "";
    },
    [t]
  ), v = n(
    (a) => {
      a.preventDefault(), r || d(!0);
    },
    [r]
  ), _ = n(() => {
    d(!1);
  }, []), k = n(
    (a) => {
      a.preventDefault(), d(!1), !r && a.dataTransfer.files.length > 0 && (t == null || t(a.dataTransfer.files));
    },
    [r, t]
  );
  return /* @__PURE__ */ l(
    "div",
    {
      className: N(
        "fds-upload-cta",
        i && "fds-upload-cta--active",
        r && "fds-upload-cta--disabled",
        g
      ),
      onDragOver: v,
      onDragLeave: _,
      onDrop: k,
      ...h,
      children: [
        /* @__PURE__ */ e(
          "div",
          {
            className: "fds-upload-cta__root",
            onClick: o,
            role: "button",
            tabIndex: r ? -1 : 0,
            "aria-disabled": r,
            onKeyDown: (a) => {
              (a.key === "Enter" || a.key === " ") && (a.preventDefault(), o());
            },
            children: /* @__PURE__ */ l("div", { className: "fds-upload-cta__inner", children: [
              /* @__PURE__ */ e("span", { className: "fds-upload-cta__body BodyMediumRegular", children: p }),
              /* @__PURE__ */ e("span", { className: "fds-upload-cta__link", children: u })
            ] })
          }
        ),
        /* @__PURE__ */ e(
          "input",
          {
            ref: f,
            type: "file",
            accept: s,
            multiple: c,
            onChange: m,
            hidden: !0,
            tabIndex: -1
          }
        )
      ]
    }
  );
}
x.displayName = "UploadCta";
export {
  x as UploadCta
};
