// @faclon-labs/design-sdk/dist/components/inputs/TextArea/TextArea.js
import { jsxs as O, jsx as s, Fragment as me } from "react/jsx-runtime";
import { forwardRef as xe, useId as ye, useState as q, useRef as Ie, useCallback as n } from "react";
import { Tag as be } from "../../data-display/Tag/Tag.js";
import { InputFieldHeader as $e } from "../../forms/InputFieldHeader/InputFieldHeader.js";
import { InputFieldFooter as he } from "../../forms/InputFieldFooter/InputFieldFooter.js";
/* empty css             */
const Ne = xe(
  ({
    label: u,
    size: f = "Medium",
    maxLines: k = 2,
    placeholder: Q,
    value: D,
    defaultValue: B,
    validationState: F = "none",
    helpText: U,
    errorText: j,
    onChange: x,
    onFocus: y,
    onBlur: I,
    isDisabled: W = !1,
    isRequired: b = !1,
    necessityIndicator: X,
    maxCharacters: d,
    isTagsMode: i = !1,
    tags: L,
    defaultTags: Y,
    onTagsChange: o,
    className: Z,
    id: g,
    name: z,
    disabled: T,
    ...M
  }, C) => {
    const ee = ye(), $ = g ?? ee, R = `${$}-help`, l = z ?? "", c = W || T || !1, V = Q ?? (i ? "Enter Description" : "Enter value"), v = X ?? (b ? "required" : void 0), [te, A] = q(!1), [re, de] = q(B ?? ""), p = D !== void 0, H = p ? D : re, [oe, K] = q(Y ?? []), [a, w] = q(""), P = Ie(null), m = L !== void 0, t = m ? L : oe, ie = (v === "required" || b) && te && (i ? t.length === 0 : !H), h = F === "error" || ie, S = h ? F === "error" ? j ?? "Error" : j ?? `${u} is required` : U ?? "", le = h ? "error" : "default", ae = H.length, se = d !== void 0 ? `${ae}/${d}` : "", G = !!S || !i && d !== void 0, J = f === "Large" ? "BodyLargeRegular" : "BodyMediumRegular", ne = n((e) => {
      const r = e.target.value;
      d !== void 0 && r.length > d || (p || de(r), x == null || x({ name: l, value: r }));
    }, [p, l, x, d]), ce = n((e) => {
      A(!0), I == null || I({ name: l, value: e.target.value });
    }, [l, I]), ue = n((e) => {
      y == null || y({ name: l, value: e.target.value });
    }, [l, y]), N = n((e) => {
      const r = e.trim();
      if (!r || t.includes(r)) return;
      const _ = [...t, r];
      m || K(_), o == null || o(_);
    }, [t, m, o]), E = n((e) => {
      const r = t.filter((_) => _ !== e);
      m || K(r), o == null || o(r);
    }, [t, m, o]), fe = n((e) => {
      e.key === "Enter" && (e.preventDefault(), N(a), w("")), e.key === "Backspace" && a === "" && t.length > 0 && E(t[t.length - 1]);
    }, [a, N, E, t]), ve = n(() => {
      A(!0), a.trim() && (N(a), w(""));
    }, [a, N]), pe = [
      "fds-textarea",
      `fds-textarea--size-${f.toLowerCase()}`,
      k > 2 && `fds-textarea--lines-${k}`,
      c && "fds-textarea--disabled",
      h && "fds-textarea--error",
      i && "fds-textarea--tags",
      Z
    ].filter(Boolean).join(" ");
    return /* @__PURE__ */ O("div", { className: pe, children: [
      /* @__PURE__ */ s(
        $e,
        {
          label: v === "optional" ? `${u} (optional)` : u,
          necessityIndicator: v === "required" ? "required" : "none",
          size: f,
          htmlFor: $
        }
      ),
      /* @__PURE__ */ s("div", { className: "fds-textarea__field-wrapper", children: /* @__PURE__ */ s(
        "div",
        {
          className: "fds-textarea__field",
          onClick: i ? () => {
            var e;
            return (e = P.current) == null ? void 0 : e.focus();
          } : void 0,
          children: i ? (
            /* —— Tags mode: chips + inline input ——————————————————————— */
            /* @__PURE__ */ O(me, { children: [
              t.map((e) => /* @__PURE__ */ s(
                be,
                {
                  id: e,
                  label: e,
                  size: f,
                  isDisabled: c,
                  onDismiss: () => E(e)
                }
              )),
              /* @__PURE__ */ s(
                "input",
                {
                  ref: P,
                  className: `fds-textarea__tag-input ${J}`,
                  id: $,
                  placeholder: t.length === 0 ? V : "",
                  value: a,
                  onChange: (e) => w(e.target.value),
                  onKeyDown: fe,
                  onBlur: ve,
                  disabled: c,
                  "aria-label": u,
                  "aria-disabled": c || void 0
                }
              )
            ] })
          ) : (
            /* —— Normal textarea mode —————————————————————————————————— */
            /* @__PURE__ */ s(
              "textarea",
              {
                ref: C,
                className: `fds-textarea__input ${J}`,
                id: $,
                name: l || void 0,
                placeholder: V,
                value: p ? D : void 0,
                defaultValue: p ? void 0 : B,
                disabled: c,
                required: v === "required" || b || void 0,
                maxLength: d,
                "aria-label": u,
                "aria-required": v === "required" || b || void 0,
                "aria-disabled": c || void 0,
                "aria-invalid": h || void 0,
                "aria-describedby": G ? R : void 0,
                onChange: ne,
                onBlur: ce,
                onFocus: ue,
                ...M
              }
            )
          )
        }
      ) }),
      G && /* @__PURE__ */ s(
        he,
        {
          helpText: S || void 0,
          counterText: !i && d !== void 0 ? se : void 0,
          state: le,
          size: f,
          id: R
        }
      )
    ] });
  }
);
Ne.displayName = "TextArea";
export {
  Ne as TextArea
};
